diff --git a/course/yui/toolboxes/toolboxes.js b/course/yui/toolboxes/toolboxes.js
index c32b441..0955095 100644
--- a/course/yui/toolboxes/toolboxes.js
+++ b/course/yui/toolboxes/toolboxes.js
@@ -370,6 +370,12 @@ YUI.add('moodle-course-toolboxes', function(Y) {
             // Prevent the default button action
             e.preventDefault();
 
+            // Return early if the current section is hidden
+            var section = e.target.ancestor(M.course.format.get_section_selector(Y));
+            if (section && section.hasClass(CSS.SECTIONHIDDENCLASS)) {
+                return;
+            }
+
             // Get the element we're working on
             var element = e.target.ancestor(CSS.ACTIVITYLI);
 
