diff --git a/login/change_password.php b/login/change_password.php
index 41d1a7d..28a4442 100644
--- a/login/change_password.php
+++ b/login/change_password.php
@@ -82,6 +82,21 @@
             print_error('errorpasswordupdate', 'auth');
         }
 
+/// Remote Learner Edit -- BEGIN
+
+        if (file_exists($CFG->dirroot . '/file/repository/repository.class.php')) {
+            require_once($CFG->dirroot . '/file/repository/repository.class.php');
+
+            if ($repo = repository_factory::factory('alfresco')) {
+                if ($repo->verify_setup() && $repo->is_configured()) {
+                    // Ensure that we push the password change out ot Alfresco.
+                    $repo->update_user_password(stripslashes_recursive($USER), $data->newpassword1);
+                }
+            }
+        }
+
+/// Remote Learner Edit -- END
+
         // register success changing password
         unset_user_preference('auth_forcepasswordchange', $USER->id);
 
