diff --git a/question/format.php b/question/format.php
index 9520ae1..cb3564a 100644
--- a/question/format.php
+++ b/question/format.php
@@ -387,7 +387,7 @@ class qformat_default {
             $question->modifiedby = $USER->id;
             $question->timemodified = time();
             $fileoptions = array(
-                    'subdirs' => true,
+                    'subdirs' => false,
                     'maxfiles' => -1,
                     'maxbytes' => 0,
                 );
