diff --git a/mod/feedback/item/multichoicerated/lib.php b/mod/feedback/item/multichoicerated/lib.php
index 0863f07..484c5f3 100644
--- a/mod/feedback/item/multichoicerated/lib.php
+++ b/mod/feedback/item/multichoicerated/lib.php
@@ -520,7 +520,6 @@ class feedback_item_multichoicerated extends feedback_item_base {
         } else {
             $hv = 'v';
         }
-        echo '<fieldset>';
         echo '<ul>';
         if (!$this->hidenoselect($item)) {
             ?>
@@ -577,7 +576,6 @@ class feedback_item_multichoicerated extends feedback_item_base {
             $index++;
         }
         echo '</ul>';
-        echo '</fieldset>';
     }
 
     private function print_item_dropdown($item, $value, $info, $align, $showrating, $lines) {
@@ -586,8 +584,10 @@ class feedback_item_multichoicerated extends feedback_item_base {
         } else {
             $hv = 'v';
         }
+        echo '<ul>';
         ?>
-        <div class="feedback_item_select_<?php echo $hv.'_'.$align;?>">
+        <li class="feedback_item_select_<?php echo $hv.'_'.$align;?>">
+            <label class="accesshide" for="<?php echo $item->typ.'_'.$item->id;?>"><?php echo $item->name; ?></label>
             <select id="<?php echo $item->typ.'_'.$item->id;?>" name="<?php echo $item->typ.'_'.$item->id;?>">
                 <option value="0">&nbsp;</option>
                 <?php
@@ -613,8 +613,9 @@ class feedback_item_multichoicerated extends feedback_item_base {
                 }
                 ?>
             </select>
-        </div>
+        </li>
         <?php
+        echo '</ul>';
     }
 
     public function prepare_presentation_values($linesep1,
