diff --git a/lib/google/curlio.php b/lib/google/curlio.php
index f3da219..654b9de 100644
--- a/lib/google/curlio.php
+++ b/lib/google/curlio.php
@@ -172,7 +172,7 @@ class moodle_google_curlio extends Google_CurlIO {
             if (!is_string($name)) {
                 $name = $this->get_option_name_from_constant($name);
             }
-            $safeParams[$name] = $value;
+            $safeParams[$name] = $vale;
         }
         parent::setOptions($safeParams);
     }
