diff --git a/blocks/activity_modules/tests/behat/block_activity_modules.feature b/blocks/activity_modules/tests/behat/block_activity_modules.feature
index 2a20f46..8883b4e 100644
--- a/blocks/activity_modules/tests/behat/block_activity_modules.feature
+++ b/blocks/activity_modules/tests/behat/block_activity_modules.feature
@@ -1,4 +1,4 @@
-@block @block_activity_modules
+@block @block_activity_modules @_only_local
 Feature: Block activity modules
   In order to overview activity modules in a course
   As a manager
@@ -15,7 +15,7 @@ Feature: Block activity modules
 
   @javascript
   Scenario: Add activities block on the frontpage
-    Given the following "activities" exists:
+    And the following "activities" exists:
       | activity   | name                        | intro                              | course               | idnumber    |
       | assign     | Frontpage assignment name   | Frontpage assignment description   | Acceptance test site | assign0     |
       | assignment | Frontpage assignment22 name | Frontpage assignment22 description | Acceptance test site | assignment0 |
