diff --git a/badges/tests/behat/add_badge.feature b/badges/tests/behat/add_badge.feature
index 2f4e162..9d4b5fa 100644
--- a/badges/tests/behat/add_badge.feature
+++ b/badges/tests/behat/add_badge.feature
@@ -1,4 +1,4 @@
-@core @core_badges @_only_local @_file_upload
+@core @core_badges @_only_local
 Feature: Add badges to the system
   In order to give badges to users for their achievements
   As an admin
