diff --git a/repository/lib.php b/repository/lib.php
index b416284..f01ccc1 100644
--- a/repository/lib.php
+++ b/repository/lib.php
@@ -2035,6 +2035,7 @@ abstract class repository {
         } else if (is_object($listing) && isset($listing->path) && is_array($listing->path)) {
             $path = &$listing->path;
         }
+
         if (isset($path)) {
             $len = count($path);
             for ($i=0; $i<$len; $i++) {
@@ -2473,6 +2474,52 @@ abstract class repository {
         $sourcefield->source = $source;
         return serialize($sourcefield);
     }
+
+    /**
+     * static method to include required CSS classes for filemanager/picker
+     * mainly for Chrome/Chromium
+     * @param object $page Optional page object, defaults to $PAGE
+     * @uses  $CFG
+     * @uses  $PAGE
+     */
+    public static function include_fmfp_css($page = null) {
+        global   $CFG, $PAGE;
+        static   $init = false;
+        if ($init) {
+            return;
+        }
+        $init = true;
+        $req_css = array(
+                     '/lib/yui/3.5.1/build/panel/assets/panel-core.css',
+                     '/lib/yui/3.5.1/build/panel/assets/skins/night/panel-skin.css',
+                     '/lib/yui/3.5.1/build/panel/assets/skins/night/panel.css',
+                     '/lib/yui/3.5.1/build/panel/assets/skins/sam/panel.css',
+                     '/lib/yui/3.5.1/build/panel/assets/skins/sam/panel-skin.css',
+                     '/lib/yui/3.5.1/build/panel/assets/skins/sam/panel.css',
+                     '/lib/yui/3.5.1/build/assets/skins/sam/skin.css'
+                   );
+
+        if (empty($page)) {
+            $page = $PAGE;
+        }
+        $out = '';
+        $headerprinted = $page->headerprinted; // not reliable!
+        foreach ($req_css as $css) {
+            if (!$headerprinted) {
+                try {
+                    $page->requires->css($css);
+                } catch (Exception $e) {
+                    $headerprinted = true;
+                }
+            }
+            if ($headerprinted) {
+                $out .= "@import url(\"{$CFG->wwwroot}{$css}\");\n";
+            }
+        }
+        if (!empty($out)) {
+            echo "<style>\n{$out}</style>";
+        }
+    }
 }
 
 /**
