diff --git a/question/type/randomsamatch/version.php b/question/type/randomsamatch/version.php
index 5e8a6b2..f5f64f7 100644
--- a/question/type/randomsamatch/version.php
+++ b/question/type/randomsamatch/version.php
@@ -17,21 +17,14 @@
 /**
  * Version information for the randomsamatch question type.
  *
- * @package    qtype_randomsamatch
+ * @package    qtype
+ * @subpackage randomsamatch
  * @copyright  1999 onwards Martin Dougiamas  {@link http://moodle.com}
  * @license    http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
  */
 
 defined('MOODLE_INTERNAL') || die();
 
-$plugin->version  = 2013050103;
+$plugin->version  = 2013050100;
 $plugin->requires = 2013050100;
-
 $plugin->component = 'qtype_randomsamatch';
-
-$plugin->dependencies = array(
-    'qtype_match' => 2013050100,
-    'qtype_shortanswer' => 2013050100,
-);
-
-$plugin->maturity  = MATURITY_STABLE;
