diff --git a/install/lang/nl/error.php b/install/lang/nl/error.php
index fae8744..3688080 100644
--- a/install/lang/nl/error.php
+++ b/install/lang/nl/error.php
@@ -46,7 +46,7 @@ $string['dmlexceptiononinstall'] = '<p>Databankfout [{$a->errorcode}].<br />{$a-
 $string['downloadedfilecheckfailed'] = 'Controle van het gedownloade bestand mislukt';
 $string['invalidmd5'] = 'De controlevariable was fout - probeer nog eens';
 $string['missingrequiredfield'] = 'Vereist veld ontbreekt';
-$string['remotedownloaderror'] = '<p>De download van de component naar jouw server is mislukt, controleer je proxy-instellingen, de PHP cURL-extentie wordt ten zeerste aanbevolen te installeren.</p><p>Je moet je <a href="{$a->url}">{$a->url}</a> bestand manueel downloaden , kopiëren naar de map "{$a->dest}" op je server en het daar unzippen.</p>';
+$string['remotedownloaderror'] = 'De download van de component naar jouw server is mislukt, controleer je proxy-instellingen, de PHP cURL-extentie wordt ten zeerste aanbevolen te installeren.<br /><br />Je moet je <a href="{$a->url}">{$a->url}</a> bestand manueel downloaden , kopiëren naar de map "{$a->dest}" op je server en het daar unzippen.';
 $string['wrongdestpath'] = 'Fout pad voor bestemming';
 $string['wrongsourcebase'] = 'Foute URL voor de bron';
 $string['wrongzipfilename'] = 'Foute ZIP-bestandsnaam';
