diff --git a/auth/tests/behat/login.feature b/auth/tests/behat/login.feature
index 793cf7e..72e4492 100644
--- a/auth/tests/behat/login.feature
+++ b/auth/tests/behat/login.feature
@@ -4,14 +4,8 @@ Feature: Authentication
   As a user
   I need to log into the system
 
-  Scenario: Login with the predefined admin user with Javascript disabled
+  Scenario: Login with the predefined admin user
     Given I log in as "admin"
-    Then I should see "You are logged in as Admin User"
-
-  @javascript
-  Scenario: Login with the predefined admin user with Javascript enabled
-    Given I log in as "admin"
-    Then I should see "You are logged in as Admin User"
 
   Scenario: Login as an existing admin user filling the form
     Given the following "users" exists:
