diff --git a/group/tests/behat/groups_import.feature b/group/tests/behat/groups_import.feature
index b8ae29e..c6a0782 100644
--- a/group/tests/behat/groups_import.feature
+++ b/group/tests/behat/groups_import.feature
@@ -1,4 +1,4 @@
-@core @core_group @_only_local @_file_upload
+@core @core_group @_only_local
 Feature: Importing of groups and groupings
   In order to import groups and grouping
   As a teacher
