diff --git a/version.php b/version.php
index 5fd841a..1b93a83 100644
--- a/version.php
+++ b/version.php
@@ -29,11 +29,11 @@
 
 defined('MOODLE_INTERNAL') || die();
 
-$version  = 2012120309.00;              // 20121203      = branching date YYYYMMDD - do not modify!
+$version  = 2012120308.00;              // 20121203      = branching date YYYYMMDD - do not modify!
                                         //         RR    = release increments - 00 in DEV branches
                                         //           .XX = incremental changes
 
-$release  = '2.4.9 (Build: 20140310)';  // Human-friendly version name
+$release  = '2.4.8 (Build: 20140113)';  // Human-friendly version name
 
 $branch   = '24';                       // this version's branch
 $maturity = MATURITY_STABLE;            // this version's maturity level
