diff --git a/question/type/randomsamatch/lang/en/qtype_randomsamatch.php b/question/type/randomsamatch/lang/en/qtype_randomsamatch.php
index 82848d9..7875885 100644
--- a/question/type/randomsamatch/lang/en/qtype_randomsamatch.php
+++ b/question/type/randomsamatch/lang/en/qtype_randomsamatch.php
@@ -23,7 +23,6 @@
  * @license    http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
  */
 
-$string['insufficientoptions'] = 'Insufficient selection options are available for this question, therefore it is not available in  this quiz. Please inform your teacher.';
 $string['nosaincategory'] = 'There are no short answer questions in the category that you chose \'{$a->catname}\'. Choose a different category, make some questions in this category.';
 $string['notenoughsaincategory'] = 'There is/are only {$a->nosaquestions} short answer questions in the category that you chose \'{$a->catname}\'. Choose a different category, make some more questions in this category or reduce the amount of questions you\'ve selected.';
 $string['pluginname'] = 'Random short-answer matching';
@@ -32,8 +31,3 @@ $string['pluginname_link'] = 'question/type/randomsamatch';
 $string['pluginnameadding'] = 'Adding a Random short-answer matching question';
 $string['pluginnameediting'] = 'Editing a Random short-answer matching question';
 $string['pluginnamesummary'] = 'Like a Matching question, but created randomly from the short answer questions in a particular category.';
-$string['randomsamatchnumber'] = 'Number of questions to select';
-$string['randomsamatch'] = 'Random short-answer matching';
-$string['randomsamatchintro'] = 'For each of the following questions, select the matching answer from the menu.';
-$string['subcats'] = 'Include subcategories';
-$string['subcats_help'] = 'If checked, questions will be choosen from subcategories too.';
