diff --git a/mod/scorm/tests/behat/add_scorm.feature b/mod/scorm/tests/behat/add_scorm.feature
index 250158f..055488e 100644
--- a/mod/scorm/tests/behat/add_scorm.feature
+++ b/mod/scorm/tests/behat/add_scorm.feature
@@ -1,4 +1,4 @@
-@mod @mod_scorm @_only_local @_file_upload
+@mod @mod_scorm @_only_local
 Feature: Add scorm activity
   In order to let students access a scorm package
   As a teacher
