diff --git a/elis/core/lib/filtering/simpleselect.php b/elis/core/lib/filtering/simpleselect.php
index 1cf1a21..5d6a964 100644
--- a/elis/core/lib/filtering/simpleselect.php
+++ b/elis/core/lib/filtering/simpleselect.php
@@ -35,11 +35,8 @@ class generalized_filter_simpleselect extends generalized_filter_type {
     /**
      * options for the list values
      */
-    var $_options = array();
-
     var $_field;
 
-    var $_options  = array();  // Select options
     var $_numeric  = false;    // TBD: obsolete - set true if field is numeric
     var $_anyvalue = null;     // The "any value" entry
     var $_noany    = false;    // Whether to hide the "any value" entry
@@ -173,7 +170,7 @@ class generalized_filter_simpleselect extends generalized_filter_type {
             }
             $a->value = implode(get_string('or', 'elis_core'), $value);
         } else {
-            $a->value = '"'.s($this->_options[$value]).'"';
+            $a->value = '"'. s($this->_options[$value]) .'"';
         }
         $a->operator = get_string('isequalto', 'filters');
 
