diff --git a/mod/choice/report.php b/mod/choice/report.php
index 2fdabd3..bd9ac97 100644
--- a/mod/choice/report.php
+++ b/mod/choice/report.php
@@ -182,7 +182,7 @@
 
         /// Print names of all the fields
 
-        echo get_string("lastname")."\t".get_string("firstname") . "\t". get_string("idnumber") . "\t";
+        echo get_string("firstname")."\t".get_string("lastname") . "\t". get_string("idnumber") . "\t";
         echo get_string("group"). "\t";
         echo get_string("choice","choice"). "\n";
 
