diff --git a/mod/glossary/deleteentry.php b/mod/glossary/deleteentry.php
index f8ca436..2449448 100644
--- a/mod/glossary/deleteentry.php
+++ b/mod/glossary/deleteentry.php
@@ -118,7 +118,7 @@ if ($confirm and confirm_sesskey()) { // the operation was confirmed.
 
 } else {        // the operation has not been confirmed yet so ask the user to do so
     $PAGE->navbar->add(get_string('delete'));
-    $PAGE->set_title($glossary->name);
+    $PAGE->set_title(format_string($glossary->name));
     $PAGE->set_heading($course->fullname);
     echo $OUTPUT->header();
     $areyousure = "<b>".format_string($entry->concept)."</b><p>$strareyousuredelete</p>";
