diff --git a/mod/glossary/editcategories.php b/mod/glossary/editcategories.php
index 47cd1e6..1c67b47 100644
--- a/mod/glossary/editcategories.php
+++ b/mod/glossary/editcategories.php
@@ -69,14 +69,15 @@ require_capability('mod/glossary:managecategories', $context);
 $strglossaries   = get_string("modulenameplural", "glossary");
 $strglossary     = get_string("modulename", "glossary");
 
-$PAGE->navbar->add(get_string("categories","glossary"),
-        new moodle_url('/mod/glossary/editcategories.php', array('id' => $cm->id,'mode' => 'cat')));
+$PAGE->navbar->add($strglossaries, new moodle_url('/mod/glossary/index.php', array('id'=>$course->id)));
+$PAGE->navbar->add(get_string("categories","glossary"));
 if (!empty($action)) {
     $navaction = get_string($action). " " . textlib::strtolower(get_string("category","glossary"));
     $PAGE->navbar->add($navaction);
 }
-$PAGE->set_title($glossary->name);
+$PAGE->set_title(format_string($glossary->name));
 $PAGE->set_heading($course->fullname);
+echo $OUTPUT->header();
 
 // Prepare format_string/text options
 $fmtoptions = array(
@@ -105,7 +106,6 @@ if ( $hook >0 ) {
             add_to_log($course->id, "glossary", "edit category", "editcategories.php?id=$cm->id", $hook,$cm->id);
 
         } else {
-            echo $OUTPUT->header();
             echo "<h3 class=\"main\">" . get_string("edit"). " " . get_string("category","glossary") . "</h3>";
 
             $name = $category->name;
@@ -120,11 +120,14 @@ if ( $hook >0 ) {
             $DB->delete_records("glossary_entries_categories", array("categoryid"=>$hook));
             $DB->delete_records("glossary_categories", array("id"=>$hook));
 
+            echo $OUTPUT->box_start('generalbox boxaligncenter errorboxcontent boxwidthnarrow');
+            echo "<div>" . get_string("categorydeleted","glossary") ."</div>";
+            echo $OUTPUT->box_end();
+
             add_to_log($course->id, "glossary", "delete category", "editcategories.php?id=$cm->id", $hook,$cm->id);
 
-            redirect("editcategories.php?id=$cm->id", get_string("categorydeleted", "glossary"), 2);
+            redirect("editcategories.php?id=$cm->id");
         } else {
-            echo $OUTPUT->header();
             echo "<p>" . get_string("delete"). " " . get_string("category","glossary"). "</p>";
 
             echo $OUTPUT->box_start('generalbox boxaligncenter errorboxcontent boxwidthnarrow');
@@ -169,7 +172,14 @@ if ( $hook >0 ) {
     if ( $confirm ) {
         $dupcategory = $DB->get_records_sql("SELECT * FROM {glossary_categories} WHERE ".$DB->sql_like('name','?', false)." AND glossaryid=?", array($name, $glossary->id));
         if ( $dupcategory ) {
-            redirect("editcategories.php?id=$cm->id&amp;action=add&amp;name=$name", get_string("duplicatecategory", "glossary"), 2);
+        echo "<h3 class=\"main\">" . get_string("add"). " " . get_string("category","glossary"). "</h3>";
+
+            echo $OUTPUT->box_start('generalbox boxaligncenter errorboxcontent boxwidthnarrow');
+            echo "<div>" . get_string("duplicatecategory","glossary") ."</div>";
+            echo $OUTPUT->box_end();
+
+            redirect("editcategories.php?id=$cm->id&amp;action=add&amp;name=$name");
+
         } else {
             $action = "";
             $cat = new stdClass();
@@ -181,7 +191,6 @@ if ( $hook >0 ) {
             add_to_log($course->id, "glossary", "add category", "editcategories.php?id=$cm->id", $cat->id,$cm->id);
         }
     } else {
-        echo $OUTPUT->header();
         echo "<h3 class=\"main\">" . get_string("add"). " " . get_string("category","glossary"). "</h3>";
         $name="";
         require "editcategories.html";
@@ -193,8 +202,6 @@ if ( $action ) {
     die;
 }
 
-echo $OUTPUT->header();
-
 ?>
 
 <form method="post" action="editcategories.php">
