diff --git a/mod/glossary/import.php b/mod/glossary/import.php
index 66a4483..388616a 100644
--- a/mod/glossary/import.php
+++ b/mod/glossary/import.php
@@ -46,7 +46,7 @@ $strsearch = get_string("search");
 $strimportentries = get_string('importentriesfromxml', 'glossary');
 
 $PAGE->navbar->add($strimportentries);
-$PAGE->set_title($glossary->name);
+$PAGE->set_title(format_string($glossary->name));
 $PAGE->set_heading($course->fullname);
 
 echo $OUTPUT->header();
@@ -98,7 +98,7 @@ if ($xml = glossary_read_imported_file($result)) {
             $glossary->showall = ($xmlglossary['SHOWALL'][0]['#']);
             $glossary->timecreated = time();
             $glossary->timemodified = time();
-            $glossary->cmidnumber = null;
+            $glossary->cmidnumber = $cm->idnumber;
 
             // Setting the default values if no values were passed
             if ( isset($xmlglossary['ENTBYPAGE'][0]['#']) ) {
@@ -132,9 +132,6 @@ if ($xml = glossary_read_imported_file($result)) {
                 $glossary->defaultapproval = $CFG->glossary_defaultapproval;
             }
 
-            // This field was not included in export, assume zero.
-            $glossary->assessed = 0;
-
             // Include new glossary and return the new ID
             if ( !$glossary->id = glossary_add_instance($glossary) ) {
                 echo $OUTPUT->notification("Error while trying to create the new glossary.");
@@ -144,7 +141,6 @@ if ($xml = glossary_read_imported_file($result)) {
             } else {
                 //The instance has been created, so lets do course_modules
                 //and course_sections
-                $mod = new stdClass();
                 $mod->groupmode = $course->groupmode;  /// Default groupmode the same as course
 
                 $mod->instance = $glossary->id;
