diff --git a/grade/tests/edittree_test.php b/grade/tests/edittree_test.php
index 1f6f65e..fab938b 100644
--- a/grade/tests/edittree_test.php
+++ b/grade/tests/edittree_test.php
@@ -52,7 +52,7 @@ class gradeedittreelib_testcase extends advanced_testcase {
         $scale = $this->getDataGenerator()->create_scale();
         $course = $this->getDataGenerator()->create_course();
         $assign = $this->getDataGenerator()->create_module('assign', array('course' => $course->id));
-        $modulecontext = context_module::instance($assign->cmid);
+        $modulecontext = context_module::instance($assign->id);
         // The generator returns a dummy object, lets get the real assign object.
         $assign = new assign($modulecontext, false, false);
         $cm = $assign->get_course_module();
