diff --git a/grade/tests/externallib_test.php b/grade/tests/externallib_test.php
index 3f6f87b..a1cbfc5 100644
--- a/grade/tests/externallib_test.php
+++ b/grade/tests/externallib_test.php
@@ -66,7 +66,7 @@ class core_grade_external_testcase extends externallib_advanced_testcase {
         // Create a teacher and give them capabilities.
         $coursecontext = context_course::instance($course->id);
         $roleid = $this->assignUserCapability('moodle/course:viewparticipants', $coursecontext->id, 3);
-        $modulecontext = context_module::instance($cm->cmid);
+        $modulecontext = context_module::instance($cm->id);
         $this->assignUserCapability('mod/assign:grade', $modulecontext->id, $roleid);
 
         // Create the teacher's enrolment record.
@@ -146,7 +146,7 @@ class core_grade_external_testcase extends externallib_advanced_testcase {
         $DB->insert_record('gradingform_rubric_levels', $rubriclevel2);
 
         // Call the external function.
-        $cmids = array ($cm->cmid);
+        $cmids = array ($cm->id);
         $areaname = 'submissions';
         $result = core_grade_external::get_definitions($cmids, $areaname);
 
