diff --git a/install/lang/sr_lt/error.php b/install/lang/sr_lt/error.php
index e11824e..262f21c 100644
--- a/install/lang/sr_lt/error.php
+++ b/install/lang/sr_lt/error.php
@@ -30,9 +30,6 @@
 
 defined('MOODLE_INTERNAL') || die();
 
-$string['cannotcreatedboninstall'] = '<p>Nije moguće kreirati bazu podataka.</p>
-<p>Specificirana baza ne postoji i dati korisnik nema prava da je kreira.</p>
-<p>Administrator sajta bi trebao da proveri podešavanja baze.</p>';
 $string['cannotcreatelangdir'] = 'Nije moguće kreirati direktorijum jezika';
 $string['cannotcreatetempdir'] = 'Nije moguće kreirati privremeni direktorijum';
 $string['cannotdownloadcomponents'] = 'Nije moguće preuzeti komponente.';
@@ -42,7 +39,6 @@ $string['cannotsavemd5file'] = 'Nije moguće sačuvati md5 datoteku.';
 $string['cannotsavezipfile'] = 'Nije moguće sačuvti ZIP arhivu.';
 $string['cannotunzipfile'] = 'Nije moguće raspakovati ZIP datoteku.';
 $string['componentisuptodate'] = 'Komponenta je dostupna u svojoj najnovijoj verziji';
-$string['dmlexceptiononinstall'] = '<p>Došlo je do greške vezane za bazu podataka [{$a->errorcode}].<br />{$a->debuginfo}</p>';
 $string['downloadedfilecheckfailed'] = 'Nije uspela provera preuzete datoteke';
 $string['invalidmd5'] = 'Neispravna md5 datoteka';
 $string['missingrequiredfield'] = 'Nedostaje neko obavezno polje';
