diff --git a/report/loglive/index.php b/report/loglive/index.php
index df365b2..782679b 100644
--- a/report/loglive/index.php
+++ b/report/loglive/index.php
@@ -44,11 +44,6 @@ $context = context_course::instance($course->id);
 require_capability('report/loglive:view', $context);
 
 $strlivelogs = get_string('livelogs', 'report_loglive');
-if (!empty($page)) {
-    $strlogs = get_string('logs'). ": ". get_string('page', 'report_loglive', $page + 1);
-} else {
-    $strlogs = get_string('logs');
-}
 
 if ($inpopup) {
     session_get_instance()->write_close();
@@ -83,7 +78,7 @@ if ($course->id == SITEID) {
 
 } else {
     $PAGE->set_url('/report/log/live.php', array('id'=>$course->id));
-    $PAGE->set_title($course->shortname .': '. $strlivelogs);
+    $PAGE->set_title($course->shortname .': '. $strlogs);
     $PAGE->set_heading($course->fullname);
     echo $OUTPUT->header();
 }
