diff --git a/question/type/edit_question_form.php b/question/type/edit_question_form.php
index bda58cc..99ef870 100644
--- a/question/type/edit_question_form.php
+++ b/question/type/edit_question_form.php
@@ -181,7 +181,7 @@ abstract class question_edit_form extends question_wizard_form {
         }
 
         $mform->addElement('text', 'name', get_string('questionname', 'question'),
-                array('size' => 50, 'maxlength' => 255));
+                array('size' => 50));
         $mform->setType('name', PARAM_TEXT);
         $mform->addRule('name', null, 'required', null, 'client');
 
