diff --git a/question/type/randomsamatch/backup/moodle2/backup_qtype_randomsamatch_plugin.class.php b/question/type/randomsamatch/backup/moodle2/backup_qtype_randomsamatch_plugin.class.php
index 2ac9191..aaa0acf 100644
--- a/question/type/randomsamatch/backup/moodle2/backup_qtype_randomsamatch_plugin.class.php
+++ b/question/type/randomsamatch/backup/moodle2/backup_qtype_randomsamatch_plugin.class.php
@@ -49,16 +49,14 @@ class backup_qtype_randomsamatch_plugin extends backup_qtype_plugin {
 
         // Now create the qtype own structures.
         $randomsamatch = new backup_nested_element('randomsamatch', array('id'), array(
-            'choose', 'subcats', 'correctfeedback', 'correctfeedbackformat',
-            'partiallycorrectfeedback', 'partiallycorrectfeedbackformat',
-            'incorrectfeedback', 'incorrectfeedbackformat', 'shownumcorrect'));
+            'choose'));
 
         // Now the own qtype tree.
         $pluginwrapper->add_child($randomsamatch);
 
         // Set source to populate the data.
-        $randomsamatch->set_source_table('qtype_randomsamatch_options',
-                array('questionid' => backup::VAR_PARENTID));
+        $randomsamatch->set_source_table('question_randomsamatch',
+                array('question' => backup::VAR_PARENTID));
 
         return $plugin;
     }
