diff --git a/course/format/lib.php b/course/format/lib.php
index acccb97..76e7649 100644
--- a/course/format/lib.php
+++ b/course/format/lib.php
@@ -237,22 +237,15 @@ abstract class format_base {
         if ($this->course === false) {
             $this->course = get_course($this->courseid);
             $options = $this->get_format_options();
-            $dbcoursecolumns = null;
             foreach ($options as $optionname => $optionvalue) {
-                if (isset($this->course->$optionname)) {
-                    // Course format options must not have the same names as existing columns in db table "course".
-                    if (!isset($dbcoursecolumns)) {
-                        $dbcoursecolumns = $DB->get_columns('course');
-                    }
-                    if (isset($dbcoursecolumns[$optionname])) {
+                if (!isset($this->course->$optionname)) {
+                    $this->course->$optionname = $optionvalue;
+                } else {
                         debugging('The option name '.$optionname.' in course format '.$this->format.
                             ' is invalid because the field with the same name exists in {course} table',
                             DEBUG_DEVELOPER);
-                        continue;
                     }
                 }
-                $this->course->$optionname = $optionvalue;
-            }
         }
         return $this->course;
     }
