diff --git a/course/format/renderer.php b/course/format/renderer.php
index 28d7bc1..85ae05f 100644
--- a/course/format/renderer.php
+++ b/course/format/renderer.php
@@ -371,8 +371,7 @@ abstract class format_section_renderer_base extends plugin_renderer_base {
                 if ($cancomplete && $completioninfo->is_enabled($thismod) != COMPLETION_TRACKING_NONE) {
                     $total++;
                     $completiondata = $completioninfo->get_data($thismod, true);
-                    if ($completiondata->completionstate == COMPLETION_COMPLETE ||
-                            $completiondata->completionstate == COMPLETION_COMPLETE_PASS) {
+                    if ($completiondata->completionstate == COMPLETION_COMPLETE) {
                         $complete++;
                     }
                 }
