diff --git a/install/lang/fr/error.php b/install/lang/fr/error.php
index 0388de7..d30b620 100644
--- a/install/lang/fr/error.php
+++ b/install/lang/fr/error.php
@@ -30,9 +30,6 @@
 
 defined('MOODLE_INTERNAL') || die();
 
-$string['cannotcreatedboninstall'] = '<p>Impossible de créer la base de données.</p>
-<p>La base de données indiquées n\'existe pas et l\'utilisateur spécifié n\'a pas les autorisations permettant de créer une base de données.</p>.
-<p>L\'administrateur du site doit revoir la configuration de la base de données.</p>';
 $string['cannotcreatelangdir'] = 'Création du dossier lang impossible';
 $string['cannotcreatetempdir'] = 'Création du dossier temp impossible';
 $string['cannotdownloadcomponents'] = 'Téléchargement des composants impossible';
@@ -42,7 +39,6 @@ $string['cannotsavemd5file'] = 'Enregistrement du fichier md5 impossible';
 $string['cannotsavezipfile'] = 'Enregistrement du fichier ZIP impossible';
 $string['cannotunzipfile'] = 'Décompression du fichier ZIP impossible';
 $string['componentisuptodate'] = 'Le composant est à jour';
-$string['dmlexceptiononinstall'] = '<p>Une erreur de base de données est survenue [{$a->errorcode}].<br />{$a->debuginfo}</p>';
 $string['downloadedfilecheckfailed'] = 'La vérification du fichier téléchargé à échoué';
 $string['invalidmd5'] = 'Le code de contrôle md5 n\'est pas valide';
 $string['missingrequiredfield'] = 'Un champ obligatoire n\'est pas renseigné';
