diff --git a/install/lang/fr/error.php b/install/lang/fr/error.php
index fed5364..b93cf32 100644
--- a/install/lang/fr/error.php
+++ b/install/lang/fr/error.php
@@ -31,8 +31,8 @@
 defined('MOODLE_INTERNAL') || die();
 
 $string['cannotcreatedboninstall'] = '<p>Impossible de créer la base de données.</p>
-<p>La base de données indiquées n\'existe pas et l\'utilisateur spécifié n\'a pas les autorisations permettant de créer une base de données.</p>.
-<p>L\'administrateur du site doit revoir la configuration de la base de données.</p>';
+<p>La base de données indiquée n\'existe pas et le nom d\'utilisateur n\'a pas l\'autorisation de créer la base de données.</p>
+<p>L\'administrateur du site doit vérifier la configuration de la base de données.</p>';
 $string['cannotcreatelangdir'] = 'Création du dossier lang impossible';
 $string['cannotcreatetempdir'] = 'Création du dossier temp impossible';
 $string['cannotdownloadcomponents'] = 'Téléchargement des composants impossible';
@@ -46,8 +46,7 @@ $string['dmlexceptiononinstall'] = '<p>Une erreur de base de données est surven
 $string['downloadedfilecheckfailed'] = 'La vérification du fichier téléchargé à échoué';
 $string['invalidmd5'] = 'Le code de contrôle md5 n\'est pas valide';
 $string['missingrequiredfield'] = 'Un champ obligatoire n\'est pas renseigné';
-$string['remotedownloaderror'] = '<p>Le téléchargement du composant sur votre serveur a échoué. Veuillez vérifier les réglages de proxy. L\'extension cURL de PHP est vivement recommandée.</p>
-<p>Vous devez télécharger manuellement le fichier <a href="{$a->url}">{$a->url}</a>, le copier sur votre serveur à l\'emplacement « {$a->dest} » et le décompresser à cet endroit.</p>';
+$string['remotedownloaderror'] = 'Le téléchargement de composants sur votre serveur a échoué. Veuillez vérifier les réglages de proxy. L\'extension cURL de PHP est vivement recommandée.<br /><br />Vous devez télécharger manuellement le fichier <a href="{$a->url}">{$a->url}</a>, le copier sur votre serveur à l\'emplacement « {$a->dest} » et le décompresser à cet endroit';
 $string['wrongdestpath'] = 'Chemin de destination incorrect';
 $string['wrongsourcebase'] = 'Adresse URL de base de la source incorrecte';
 $string['wrongzipfilename'] = 'Nom de fichier ZIP incorrect';
