diff --git a/cohort/tests/behat/behat_cohort.php b/cohort/tests/behat/behat_cohort.php
index f39a626..5349c53 100644
--- a/cohort/tests/behat/behat_cohort.php
+++ b/cohort/tests/behat/behat_cohort.php
@@ -17,7 +17,7 @@
 /**
  * Cohorts steps definitions.
  *
- * @package    core_cohort
+ * @package    core
  * @category   test
  * @copyright  2013 David Monllaó
  * @license    http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
@@ -32,7 +32,7 @@ use Behat\Behat\Context\Step\Given as Given;
 /**
  * Steps definitions for cohort actions.
  *
- * @package    core_cohort
+ * @package    core
  * @category   test
  * @copyright  2013 David Monllaó
  * @license    http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
