diff --git a/cohort/locallib.php b/cohort/locallib.php
index 62dcfdf..23fa8a6 100644
--- a/cohort/locallib.php
+++ b/cohort/locallib.php
@@ -126,7 +126,7 @@ class cohort_existing_selector extends user_selector_base {
 
         if (!$this->is_validating()) {
             $potentialmemberscount = $DB->count_records_sql($countfields . $sql, $params);
-            if ($potentialmemberscount > $this->maxusersperpage) {
+            if ($potentialmemberscount > 100) {
                 return $this->too_many_results($search, $potentialmemberscount);
             }
         }
