diff --git a/blocks/tests/behat/behat_blocks.php b/blocks/tests/behat/behat_blocks.php
index 215c72e..42fe9a1 100644
--- a/blocks/tests/behat/behat_blocks.php
+++ b/blocks/tests/behat/behat_blocks.php
@@ -17,7 +17,7 @@
 /**
  * Steps definitions related with blocks.
  *
- * @package   core_block
+ * @package   core
  * @category  test
  * @copyright 2012 David Monllaó
  * @license   http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
@@ -32,7 +32,7 @@ use Behat\Behat\Context\Step\Given as Given;
 /**
  * Blocks management steps definitions.
  *
- * @package    core_block
+ * @package    core
  * @category   test
  * @copyright  2012 David Monllaó
  * @license    http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
