diff --git a/badges/view.php b/badges/view.php
index 38c0fe9..d623c81 100644
--- a/badges/view.php
+++ b/badges/view.php
@@ -73,6 +73,11 @@ if ($type == BADGE_TYPE_SITE) {
     $PAGE->set_context(context_course::instance($course->id));
     $PAGE->set_pagelayout('course');
     $PAGE->set_heading($title);
+
+    // Fix course navigation.
+    $PAGE->navbar->ignore_active();
+    $PAGE->navbar->add($course->shortname, new moodle_url('/course/view.php', array('id' => $course->id)));
+    $PAGE->navbar->add(get_string('coursebadges', 'badges'));
 }
 
 $PAGE->set_title($title);
