diff --git a/lib/yui/build/moodle-core-tooltip/moodle-core-tooltip-debug.js b/lib/yui/build/moodle-core-tooltip/moodle-core-tooltip-debug.js
index 70005b8..1071b0b 100644
--- a/lib/yui/build/moodle-core-tooltip/moodle-core-tooltip-debug.js
+++ b/lib/yui/build/moodle-core-tooltip/moodle-core-tooltip-debug.js
@@ -422,7 +422,7 @@ Y.extend(TOOLTIP, M.core.dialogue, {
 
     close_panel: function(e) {
         // Hide the panel first.
-        this.hide(e);
+        this.hide();
 
         // Cancel the listeners that we added in display_panel.
         this.cancel_events();
