diff --git a/lib/ajax/blocks.php b/lib/ajax/blocks.php
index e78b756..ab24525 100644
--- a/lib/ajax/blocks.php
+++ b/lib/ajax/blocks.php
@@ -88,14 +88,9 @@ switch ($action) {
 
         $bui_newweight = null;
         if ($bui_beforeid == 0) {
-            if (count($instances) === 0) {
-                // Moving the block into an empty region. Give it the default weight.
-                $bui_newweight = 0;
-            } else {
-                // Moving to very bottom.
+            // Moving to very bottom
                 $last = end($instances);
                 $bui_newweight = $last->instance->weight + 1;
-            }
         } else {
             // Moving somewhere
             $lastweight = 0;
