diff --git a/lib/ajax/getnavbranch.php b/lib/ajax/getnavbranch.php
index 457f8d9..957d0a9 100644
--- a/lib/ajax/getnavbranch.php
+++ b/lib/ajax/getnavbranch.php
@@ -112,16 +112,16 @@ try {
     $html = ob_get_contents();
     ob_end_clean();
 } catch (Exception $e) {
-    throw new coding_exception('Error: '.$e->getMessage());
+    die('Error: '.$e->getMessage());
 }
 
 // Check if the buffer contianed anything if it did ERROR!
 if (trim($html) !== '') {
-    throw new coding_exception('Errors were encountered while producing the navigation branch'."\n\n\n".$html);
+    die('Errors were encountered while producing the navigation branch'."\n\n\n".$html);
 }
 // Check that branch isn't empty... if it is ERROR!
-if (empty($branch) || ($branch->nodetype !== navigation_node::NODETYPE_BRANCH && !$branch->isexpandable)) {
-    throw new coding_exception('No further information available for this branch');
+if (empty($branch) || $branch->nodetype !== navigation_node::NODETYPE_BRANCH) {
+    die('No further information available for this branch');
 }
 
 // Prepare an XML converter for the branch
