diff --git a/lib/phpunit/classes/hint_resultprinter.php b/lib/phpunit/classes/hint_resultprinter.php
index 2d5dfe7..a04b326 100644
--- a/lib/phpunit/classes/hint_resultprinter.php
+++ b/lib/phpunit/classes/hint_resultprinter.php
@@ -108,11 +108,9 @@ class Hint_ResultPrinter extends PHPUnit_TextUI_ResultPrinter {
             $executable = 'phpunit';
             if (testing_is_cygwin()) {
                 $file = str_replace('\\', '/', $file);
-                if (!testing_is_mingw()) {
                     $executable = 'phpunit.bat';
                 }
             }
-        }
 
         $this->write("\nTo re-run:\n $executable $testName $file\n");
     }
