diff --git a/mod/quiz/editlib.php b/mod/quiz/editlib.php
index b395d91..6cebed7 100644
--- a/mod/quiz/editlib.php
+++ b/mod/quiz/editlib.php
@@ -1018,7 +1018,7 @@ function quiz_question_tostring($question, $showicon = false,
         $questiontext = shorten_text($questiontext, 200);
         $result .= '<span class="questiontext">';
         if (!empty($questiontext)) {
-            $result .= s($questiontext);
+            $result .= $questiontext;
         } else {
             $result .= '<span class="error">';
             $result .= get_string('questiontextisempty', 'quiz');
