diff --git a/mod/quiz/override_form.php b/mod/quiz/override_form.php
index 2dd60cf..80976ec 100644
--- a/mod/quiz/override_form.php
+++ b/mod/quiz/override_form.php
@@ -27,7 +27,6 @@
 defined('MOODLE_INTERNAL') || die();
 
 require_once($CFG->libdir . '/formslib.php');
-require_once($CFG->dirroot . '/mod/quiz/mod_form.php');
 
 
 /**
@@ -158,15 +157,10 @@ class quiz_override_form extends moodleform {
                 }
 
                 $userchoices = array();
-                $canviewemail = in_array('email', get_extra_user_fields($this->context));
                 foreach ($users as $id => $user) {
                     if (empty($invalidusers[$id]) || (!empty($override) &&
                             $id == $override->userid)) {
-                        if ($canviewemail) {
                             $userchoices[$id] = fullname($user) . ', ' . $user->email;
-                        } else {
-                            $userchoices[$id] = fullname($user);
-                        }
                     }
                 }
                 unset($users);
@@ -190,11 +184,11 @@ class quiz_override_form extends moodleform {
 
         // Open and close dates.
         $mform->addElement('date_time_selector', 'timeopen',
-                get_string('quizopen', 'quiz'), mod_quiz_mod_form::$datefieldoptions);
+                get_string('quizopen', 'quiz'), array('optional' => true));
         $mform->setDefault('timeopen', $this->quiz->timeopen);
 
         $mform->addElement('date_time_selector', 'timeclose',
-                get_string('quizclose', 'quiz'), mod_quiz_mod_form::$datefieldoptions);
+                get_string('quizclose', 'quiz'), array('optional' => true));
         $mform->setDefault('timeclose', $this->quiz->timeclose);
 
         // Time limit.
