diff --git a/mod/quiz/review.php b/mod/quiz/review.php
index 9c63cd9..5ef1b94 100644
--- a/mod/quiz/review.php
+++ b/mod/quiz/review.php
@@ -103,7 +103,7 @@ if ($attemptobj->is_preview_user() && $attemptobj->is_own_attempt()) {
 
 // Set up the page header.
 $headtags = $attemptobj->get_html_head_contributions($page, $showall);
-$PAGE->set_title($attemptobj->get_quiz_name());
+$PAGE->set_title(format_string($attemptobj->get_quiz_name()));
 $PAGE->set_heading($attemptobj->get_course()->fullname);
 
 // Summary table start. ============================================================================
