diff --git a/question/engine/bank.php b/question/engine/bank.php
index dc8a8a6..b329b31 100644
--- a/question/engine/bank.php
+++ b/question/engine/bank.php
@@ -435,6 +435,9 @@ class question_finder implements cache_data_source {
     /** @var question_finder the singleton instance of this class. */
     protected static $questionfinder = null;
 
+    /** @var cache the question definition cache. */
+    protected $cache = null;
+
     /**
      * @return question_finder a question finder.
      */
@@ -454,8 +457,10 @@ class question_finder implements cache_data_source {
      * @return get the question definition cache we are using.
      */
     protected function get_data_cache() {
-        // Do not double cache here because it may break cache resetting.
-        return cache::make('core', 'questiondata');
+        if ($this->cache == null) {
+            $this->cache = cache::make('core', 'questiondata');
+        }
+        return $this->cache;
     }
 
     /**
