diff --git a/enrol/imsenterprise/importnow.php b/enrol/imsenterprise/importnow.php
index 078e899..102d038 100644
--- a/enrol/imsenterprise/importnow.php
+++ b/enrol/imsenterprise/importnow.php
@@ -2,7 +2,6 @@
 require_once(dirname(dirname(dirname(__FILE__))) . '/config.php');
 require_login(0, false);
 require_capability('moodle/site:config', context_system::instance());
-require_sesskey();
 
 $site = get_site();
 
