diff --git a/enrol/imsenterprise/settings.php b/enrol/imsenterprise/settings.php
index c1e13b5..5b7fbb6 100644
--- a/enrol/imsenterprise/settings.php
+++ b/enrol/imsenterprise/settings.php
@@ -94,8 +94,6 @@ if ($ADMIN->fulltree) {
 
     $settings->add(new admin_setting_configcheckbox('enrol_imsenterprise/imscapitafix', get_string('usecapitafix', 'enrol_imsenterprise'), get_string('usecapitafix_desc', 'enrol_imsenterprise'), 0));
 
-    $importurl = new moodle_url('/enrol/imsenterprise/importnow.php', array('sesskey' => sesskey()));
-    $importnowstring = get_string('aftersaving...', 'enrol_imsenterprise').' '.
-        html_writer::link($importurl, get_string('doitnow', 'enrol_imsenterprise'));
+    $importnowstring = get_string('aftersaving...', 'enrol_imsenterprise').' <a href="../enrol/imsenterprise/importnow.php">'.get_string('doitnow', 'enrol_imsenterprise').'</a>';
     $settings->add(new admin_setting_heading('enrol_imsenterprise_doitnowmessage', '', $importnowstring));
 }
