diff --git a/repository/tests/repository_test.php b/repository/tests/repository_test.php
index b30f0a4..9b4c8ae 100644
--- a/repository/tests/repository_test.php
+++ b/repository/tests/repository_test.php
@@ -246,10 +246,10 @@ class repositorylib_testcase extends advanced_testcase {
         $forumdata = new stdClass();
         $forumdata->course = $course1->id;
         $forumc1 = $this->getDataGenerator()->create_module('forum', $forumdata);
-        $forumc1context = context_module::instance($forumc1->cmid);
+        $forumc1context = context_module::instance($forumc1->id);
         $forumdata->course = $course2->id;
         $forumc2 = $this->getDataGenerator()->create_module('forum', $forumdata);
-        $forumc2context = context_module::instance($forumc2->cmid);
+        $forumc2context = context_module::instance($forumc2->id);
 
         $blockdata = new stdClass();
         $blockdata->parentcontextid = $course1context->id;
