diff --git a/question/type/match/renderer.php b/question/type/match/renderer.php
index 09c2c7a..5bc2f82 100644
--- a/question/type/match/renderer.php
+++ b/question/type/match/renderer.php
@@ -58,7 +58,9 @@ class qtype_match_renderer extends qtype_with_combined_feedback_renderer {
             $result .= html_writer::start_tag('tr', array('class' => 'r' . $parity));
             $fieldname = 'sub' . $key;
 
-            $result .= html_writer::tag('td', $this->format_stem_text($qa, $stemid),
+            $result .= html_writer::tag('td', $question->format_text(
+                    $question->stems[$stemid], $question->stemformat[$stemid],
+                    $qa, 'qtype_match', 'subquestion', $stemid),
                     array('class' => 'text'));
 
             $classes = 'control';
@@ -107,20 +109,6 @@ class qtype_match_renderer extends qtype_with_combined_feedback_renderer {
         return $this->combined_feedback($qa);
     }
 
-    /**
-     * Format each question stem. Overwritten by randomsamatch renderer.
-     *
-     * @param question_attempt $qa
-     * @param integer $stemid stem index
-     * @return string
-     */
-    public function format_stem_text($qa, $stemid) {
-        $question = $qa->get_question();
-        return $question->format_text(
-                    $question->stems[$stemid], $question->stemformat[$stemid],
-                    $qa, 'qtype_match', 'subquestion', $stemid);
-    }
-
     protected function format_choices($question) {
         $choices = array();
         foreach ($question->get_choice_order() as $key => $choiceid) {
@@ -137,7 +125,9 @@ class qtype_match_renderer extends qtype_with_combined_feedback_renderer {
         $choices = $this->format_choices($question);
         $right = array();
         foreach ($stemorder as $key => $stemid) {
-            $right[] = $this->format_stem_text($qa, $stemid) . ' – ' .
+            $right[] = $question->format_text($question->stems[$stemid],
+                    $question->stemformat[$stemid], $qa,
+                    'qtype_match', 'subquestion', $stemid) . ' – ' .
                     $choices[$question->get_right_choice_for($stemid)];
         }
 
