diff --git a/mod/feedback/complete_guest.php b/mod/feedback/complete_guest.php
index c9a7684..695cf41 100644
--- a/mod/feedback/complete_guest.php
+++ b/mod/feedback/complete_guest.php
@@ -162,8 +162,8 @@ $PAGE->set_pagelayout('incourse');
 $urlparams = array('id'=>$course->id);
 $PAGE->navbar->add($strfeedbacks, new moodle_url('/mod/feedback/index.php', $urlparams));
 $PAGE->navbar->add(format_string($feedback->name));
-$PAGE->set_heading($course->fullname);
-$PAGE->set_title($feedback->name);
+$PAGE->set_heading(format_string($course->fullname));
+$PAGE->set_title(format_string($feedback->name));
 echo $OUTPUT->header();
 
 //ishidden check.
@@ -177,8 +177,8 @@ if ((empty($cm->visible) AND
 //check, if the feedback is open (timeopen, timeclose)
 $checktime = time();
 
-$feedback_is_closed = ($feedback->timeopen > $checktime) ||
-                      ($feedback->timeclose < $checktime &&
+$feedback_is_closed = ($feedback->timeopen > $checktime) OR
+                      ($feedback->timeclose < $checktime AND
                             $feedback->timeclose > 0);
 
 if ($feedback_is_closed) {
@@ -386,6 +386,7 @@ if ($feedback_can_submit) {
         if (is_array($feedbackitems)) {
             echo $OUTPUT->box_start('feedback_form');
             echo '<form action="complete_guest.php" method="post" onsubmit=" ">';
+            echo '<fieldset>';
             echo '<input type="hidden" name="anonymous" value="0" />';
             $inputvalue = 'value="'.FEEDBACK_ANONYMOUS_YES.'"';
             echo '<input type="hidden" name="anonymous_response" '.$inputvalue.' />';
@@ -500,6 +501,7 @@ if ($feedback_can_submit) {
                 echo '<input name="savevalues" type="submit" '.$inputvalue.' />';
             }
 
+            echo '</fieldset>';
             echo '</form>';
             echo $OUTPUT->box_end();
 
@@ -514,9 +516,11 @@ if ($feedback_can_submit) {
                 }
             }
             echo '<form '.$action.' method="post" onsubmit=" ">';
+            echo '<fieldset>';
             echo '<input type="hidden" name="sesskey" value="'.sesskey().'" />';
             echo '<input type="hidden" name="courseid" value="'. $courseid . '" />';
             echo '<button type="submit">'.get_string('cancel').'</button>';
+            echo '</fieldset>';
             echo '</form>';
             echo $OUTPUT->box_end();
             $SESSION->feedback->is_started = true;
