diff --git a/mod/feedback/complete.php b/mod/feedback/complete.php
index e41c845..d4dc00b 100644
--- a/mod/feedback/complete.php
+++ b/mod/feedback/complete.php
@@ -176,8 +176,8 @@ if ((empty($cm->visible) AND
 
 //check, if the feedback is open (timeopen, timeclose)
 $checktime = time();
-$feedback_is_closed = ($feedback->timeopen > $checktime) ||
-                      ($feedback->timeclose < $checktime &&
+$feedback_is_closed = ($feedback->timeopen > $checktime) OR
+                      ($feedback->timeclose < $checktime AND
                             $feedback->timeclose > 0);
 
 if ($feedback_is_closed) {
