diff --git a/mod/feedback/delete_template.php b/mod/feedback/delete_template.php
index ddd5b03..289f208 100644
--- a/mod/feedback/delete_template.php
+++ b/mod/feedback/delete_template.php
@@ -107,8 +107,8 @@ $strfeedbacks = get_string("modulenameplural", "feedback");
 $strfeedback  = get_string("modulename", "feedback");
 $strdeletefeedback = get_string('delete_template', 'feedback');
 
-$PAGE->set_heading($course->fullname);
-$PAGE->set_title($feedback->name);
+$PAGE->set_heading(format_string($course->fullname));
+$PAGE->set_title(format_string($feedback->name));
 echo $OUTPUT->header();
 
 /// print the tabs
