diff --git a/mod/feedback/delete_completed.php b/mod/feedback/delete_completed.php
index fc17a5f..d5eacb4 100644
--- a/mod/feedback/delete_completed.php
+++ b/mod/feedback/delete_completed.php
@@ -96,8 +96,8 @@ $strfeedbacks = get_string("modulenameplural", "feedback");
 $strfeedback  = get_string("modulename", "feedback");
 
 $PAGE->navbar->add(get_string('delete_entry', 'feedback'));
-$PAGE->set_heading($course->fullname);
-$PAGE->set_title($feedback->name);
+$PAGE->set_heading(format_string($course->fullname));
+$PAGE->set_title(format_string($feedback->name));
 echo $OUTPUT->header();
 
 /// Print the main part of the page
