diff --git a/mod/feedback/delete_item.php b/mod/feedback/delete_item.php
index 7cdb144..d0840fb 100644
--- a/mod/feedback/delete_item.php
+++ b/mod/feedback/delete_item.php
@@ -71,8 +71,8 @@ $strfeedbacks = get_string("modulenameplural", "feedback");
 $strfeedback  = get_string("modulename", "feedback");
 
 $PAGE->navbar->add(get_string('delete_item', 'feedback'));
-$PAGE->set_heading($course->fullname);
-$PAGE->set_title($feedback->name);
+$PAGE->set_heading(format_string($course->fullname));
+$PAGE->set_title(format_string($feedback->name));
 echo $OUTPUT->header();
 
 /// Print the main part of the page
