diff --git a/mod/feedback/edit.php b/mod/feedback/edit.php
index 34fbeac..cdab171 100644
--- a/mod/feedback/edit.php
+++ b/mod/feedback/edit.php
@@ -155,8 +155,8 @@ $strfeedbacks = get_string('modulenameplural', 'feedback');
 $strfeedback  = get_string('modulename', 'feedback');
 
 $PAGE->set_url('/mod/feedback/edit.php', array('id'=>$cm->id, 'do_show'=>$do_show));
-$PAGE->set_heading($course->fullname);
-$PAGE->set_title($feedback->name);
+$PAGE->set_heading(format_string($course->fullname));
+$PAGE->set_title(format_string($feedback->name));
 
 //Adding the javascript module for the items dragdrop.
 if (count($feedbackitems) > 1) {
