diff --git a/mod/feedback/import.php b/mod/feedback/import.php
index d6d9f98..7733292 100644
--- a/mod/feedback/import.php
+++ b/mod/feedback/import.php
@@ -92,8 +92,8 @@ if ($choosefile) {
 $strfeedbacks = get_string("modulenameplural", "feedback");
 $strfeedback  = get_string("modulename", "feedback");
 
-$PAGE->set_heading($course->fullname);
-$PAGE->set_title($feedback->name);
+$PAGE->set_heading(format_string($course->fullname));
+$PAGE->set_title(format_string($feedback->name));
 echo $OUTPUT->header();
 
 /// print the tabs
