diff --git a/mod/feedback/print.php b/mod/feedback/print.php
index 18bbde9..ac2ed30 100644
--- a/mod/feedback/print.php
+++ b/mod/feedback/print.php
@@ -56,8 +56,8 @@ $feedback_url = new moodle_url('/mod/feedback/index.php', array('id'=>$course->i
 $PAGE->navbar->add($strfeedbacks, $feedback_url);
 $PAGE->navbar->add(format_string($feedback->name));
 
-$PAGE->set_title($feedback->name);
-$PAGE->set_heading($course->fullname);
+$PAGE->set_title(format_string($feedback->name));
+$PAGE->set_heading(format_string($course->fullname));
 echo $OUTPUT->header();
 
 /// Print the main part of the page
