diff --git a/mod/feedback/show_entries_anonym.php b/mod/feedback/show_entries_anonym.php
index c189e37..bf69bf3 100644
--- a/mod/feedback/show_entries_anonym.php
+++ b/mod/feedback/show_entries_anonym.php
@@ -62,8 +62,8 @@ require_capability('mod/feedback:viewreports', $context);
 $strfeedbacks = get_string("modulenameplural", "feedback");
 $strfeedback  = get_string("modulename", "feedback");
 
-$PAGE->set_heading($course->fullname);
-$PAGE->set_title($feedback->name);
+$PAGE->set_heading(format_string($course->fullname));
+$PAGE->set_title(format_string($feedback->name));
 echo $OUTPUT->header();
 
 /// Print the main part of the page
