diff --git a/mod/feedback/show_nonrespondents.php b/mod/feedback/show_nonrespondents.php
index b496fbf..4a95584 100644
--- a/mod/feedback/show_nonrespondents.php
+++ b/mod/feedback/show_nonrespondents.php
@@ -131,8 +131,8 @@ if ($action == 'sendmessage' AND has_capability('moodle/course:bulkmessaging', $
 
 /// Print the page header
 $PAGE->navbar->add(get_string('show_nonrespondents', 'feedback'));
-$PAGE->set_heading($course->fullname);
-$PAGE->set_title($feedback->name);
+$PAGE->set_heading(format_string($course->fullname));
+$PAGE->set_title(format_string($feedback->name));
 echo $OUTPUT->header();
 
 require('tabs.php');
