diff --git a/repository/alfresco/version.php b/repository/alfresco/version.php
index 10aa7cf..964e80b 100644
--- a/repository/alfresco/version.php
+++ b/repository/alfresco/version.php
@@ -26,6 +26,6 @@
 
 defined('MOODLE_INTERNAL') || die();
 
-$plugin->version   = 2014020301;        // The current plugin version (Date: YYYYMMDDXX)
+$plugin->version   = 2013050100;        // The current plugin version (Date: YYYYMMDDXX)
 $plugin->requires  = 2013050100;        // Requires this Moodle version
 $plugin->component = 'repository_alfresco'; // Full name of the plugin (used for diagnostics)
