diff --git a/question/behaviour/behaviourbase.php b/question/behaviour/behaviourbase.php
index 44de639..4f3e82c 100644
--- a/question/behaviour/behaviourbase.php
+++ b/question/behaviour/behaviourbase.php
@@ -229,7 +229,7 @@ abstract class question_behaviour {
             return array();
         }
 
-        $vars = array('comment' => PARAM_RAW, 'commentformat' => PARAM_INT);
+        $vars = array('comment' => PARAM_RAW);
         if ($this->qa->get_max_mark()) {
             $vars['mark'] = question_attempt::PARAM_MARK;
             $vars['maxmark'] = PARAM_FLOAT;
