diff --git a/question/engine/tests/helpers.php b/question/engine/tests/helpers.php
index 4977b65..3113cb3 100644
--- a/question/engine/tests/helpers.php
+++ b/question/engine/tests/helpers.php
@@ -237,7 +237,7 @@ class test_question_maker {
             array($qtype,    $which), $methodtemplate);
 
         if (!method_exists($helper, $method)) {
-            throw new coding_exception('Method ' . $method . ' does not exist on the ' .
+            throw new coding_exception('Method ' . $method . ' does not exist on the' .
                 $qtype . ' question type test helper class.');
         }
 
@@ -824,22 +824,13 @@ abstract class qbehaviour_walkthrough_test_base extends question_testcase {
                 'Looking for a hidden input with attributes ' . html_writer::attributes($attributes) . ' in ' . $this->currentoutput);
     }
 
-    protected function check_output_contains($string) {
+    protected function check_output_contains_lang_string($identifier, $component = '', $a = null) {
         $this->render();
+        $string = get_string($identifier, $component, $a);
         $this->assertContains($string, $this->currentoutput,
                 'Expected string ' . $string . ' not found in ' . $this->currentoutput);
     }
 
-    protected function check_output_does_not_contain($string) {
-        $this->render();
-        $this->assertNotContains($string, $this->currentoutput,
-                'String ' . $string . ' unexpectedly found in ' . $this->currentoutput);
-    }
-
-    protected function check_output_contains_lang_string($identifier, $component = '', $a = null) {
-        $this->check_output_contains(get_string($identifier, $component, $a));
-    }
-
     protected function get_tag_matcher($tag, $attributes) {
         return array(
             'tag' => $tag,
