diff --git a/mod/scorm/version.php b/mod/scorm/version.php
index a778141..1ca0c5d 100644
--- a/mod/scorm/version.php
+++ b/mod/scorm/version.php
@@ -17,14 +17,15 @@
 /**
  * scorm version information.
  *
- * @package    mod_scorm
+ * @package    mod
+ * @subpackage scorm
  * @copyright  1999 onwards Martin Dougiamas  {@link http://moodle.com}
  * @license    http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
  */
 
 defined('MOODLE_INTERNAL') || die();
 
-$module->version   = 2013050104;       // The current module version (Date: YYYYMMDDXX)
+$module->version   = 2013050103;       // The current module version (Date: YYYYMMDDXX)
 $module->requires  = 2013050100;    // Requires this Moodle version
 $module->component = 'mod_scorm'; // Full name of the plugin (used for diagnostics)
 $module->cron      = 300;
