diff --git a/repository/tests/behat/delete_files.feature b/repository/tests/behat/delete_files.feature
index 0d1fdc6..c6bb5ff 100644
--- a/repository/tests/behat/delete_files.feature
+++ b/repository/tests/behat/delete_files.feature
@@ -1,4 +1,4 @@
-@core @core_filepicker @_only_local @_file_upload
+@core @core_filepicker @_only_local
 Feature: Delete files and folders from the file manager
   In order to clean the file manager contents
   As a user
