diff --git a/mod/assign/gradingtable.php b/mod/assign/gradingtable.php
index 08a1de1..57df5a0 100644
--- a/mod/assign/gradingtable.php
+++ b/mod/assign/gradingtable.php
@@ -605,9 +605,6 @@ class assign_grading_table extends table_sql implements renderable {
                               id="selectuser_' . $row->userid . '"
                               name="selectedusers"
                               value="' . $row->userid . '"/>';
-        $selectcol .= '<input type="hidden"
-                              name="grademodified_' . $row->userid . '"
-                              value="' . $row->timemarked . '"/>';
         return $selectcol;
     }
 
