diff --git a/user/editadvanced.php b/user/editadvanced.php
index 74c1709..4c37e01 100644
--- a/user/editadvanced.php
+++ b/user/editadvanced.php
@@ -119,6 +119,22 @@
                         error('Failed to update password on external auth: ' . $usernew->auth .
                                 '. See the server logs for more details.');
                     }
+
+/// Remote Learner Edit -- BEGIN
+
+                    if (file_exists($CFG->dirroot . '/file/repository/repository.class.php')) {
+                        require_once($CFG->dirroot . '/file/repository/repository.class.php');
+
+                        if ($repo = repository_factory::factory('alfresco')) {
+                            if ($repo->verify_setup() && $repo->is_configured()) {
+                                // Ensure that we push the password change out ot Alfresco.
+                                $repo->update_user_password(stripslashes_recursive($usernew), $usernew->newpassword);
+                            }
+                        }
+                    }
+
+/// Remote Learner Edit -- END
+
                 }
             }
             $usercreated = false;
