diff --git a/user/files.php b/user/files.php
index 54f7962..7b4a7c3 100644
--- a/user/files.php
+++ b/user/files.php
@@ -26,6 +26,7 @@
 require('../config.php');
 require_once("$CFG->dirroot/user/files_form.php");
 require_once("$CFG->dirroot/repository/lib.php");
+require_once($CFG->dirroot.'/repository/elis_files/lib/lib.php');
 
 require_login();
 if (isguestuser()) {
@@ -33,6 +34,8 @@ if (isguestuser()) {
 }
 
 $returnurl = optional_param('returnurl', '', PARAM_URL);
+$courseid = optional_param('courseid', SITEID, PARAM_INT); // RL EDIT: ELIS-7127
+$returnbutton = true;
 
 if (empty($returnurl)) {
     $returnurl = new moodle_url('/user/files.php');
@@ -51,9 +54,21 @@ $PAGE->set_heading($title);
 $PAGE->set_pagelayout('mydashboard');
 $PAGE->set_pagetype('user-files');
 
+// RL edit: need this module for the rendering of the ELIS Files advanced search UI
+$PAGE->requires->yui2_lib(array('layout', 'container', 'dragdrop'));
+
+// Obtain the UUID for the default browsing location
+$currentpath = elis_files_get_current_path_for_course($courseid, true);
+
 $data = new stdClass();
 $data->returnurl = $returnurl;
-$options = array('subdirs'=>1, 'maxbytes'=>$CFG->userquota, 'maxfiles'=>-1, 'accepted_types'=>'*');
+$options = array(
+    'subdirs'        => 1,
+    'maxbytes'       => $CFG->userquota,
+    'maxfiles'       => -1,
+    'accepted_types' => '*',
+    'currentpath'    => $currentpath
+);
 file_prepare_standard_filemanager($data, 'files', $options, $context, 'user', 'private', 0);
 
 $mform = new user_files_form(null, array('data'=>$data, 'options'=>$options));
