diff --git a/admin/tests/behat/behat_admin.php b/admin/tests/behat/behat_admin.php
index ec3da56..a1cbcd1 100644
--- a/admin/tests/behat/behat_admin.php
+++ b/admin/tests/behat/behat_admin.php
@@ -17,7 +17,7 @@
 /**
  * Steps definitions related with administration.
  *
- * @package   core_admin
+ * @package   core
  * @category  test
  * @copyright 2013 David Monllaó
  * @license   http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
@@ -34,7 +34,7 @@ use Behat\Gherkin\Node\TableNode as TableNode,
 /**
  * Site administration level steps definitions.
  *
- * @package    core_admin
+ * @package    core
  * @category   test
  * @copyright  2013 David Monllaó
  * @license    http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
