diff --git a/question/engine/upgrade/upgradelib.php b/question/engine/upgrade/upgradelib.php
index 9cb15e5..531889d 100644
--- a/question/engine/upgrade/upgradelib.php
+++ b/question/engine/upgrade/upgradelib.php
@@ -621,32 +621,3 @@ class question_deleted_question_attempt_updater extends question_qtype_attempt_u
         $data['upgradedfromdeletedquestion'] = $state->answer;
     }
 }
-
-/**
- * This check verifies that all quiz attempts were upgraded since following
- * the question engine upgrade in Moodle 2.1.
- *
- * @param environment_results object to update, if relevant.
- * @return environment_results updated results object, or null if this test is not relevant.
- */
-function quiz_attempts_upgraded(environment_results $result) {
-    global $DB;
-
-    $dbman = $DB->get_manager();
-    $table = new xmldb_table('quiz_attempts');
-    $field = new xmldb_field('needsupgradetonewqe');
-
-    if (!$dbman->table_exists($table) || !$dbman->field_exists($table, $field)) {
-        // DB already upgraded. This test is no longer relevant.
-        return null;
-    }
-
-    if (!$DB->record_exists('quiz_attempts', array('needsupgradetonewqe' => 1))) {
-        // No 1s present in that column means there are no problems.
-        return null;
-    }
-
-    // Only display anything if the admins need to be aware of the problem.
-    $result->setStatus(false);
-    return $result;
-}
