diff --git a/install/lang/sr_cr/error.php b/install/lang/sr_cr/error.php
index 467caa7..92050fc 100644
--- a/install/lang/sr_cr/error.php
+++ b/install/lang/sr_cr/error.php
@@ -30,9 +30,6 @@
 
 defined('MOODLE_INTERNAL') || die();
 
-$string['cannotcreatedboninstall'] = '<p>Није могуће креирати базу података.</p>
-<p>Специфицирана база не постоји и дати корисник нема права да је креира.</p>
-<p>Администратор сајта би требао да провери подешавања базе.</p>';
 $string['cannotcreatelangdir'] = 'Није могуће креирати директоријум језика';
 $string['cannotcreatetempdir'] = 'Није могуће креирати привремени директоријум';
 $string['cannotdownloadcomponents'] = 'Није могуће преузети компоненте.';
@@ -42,7 +39,6 @@ $string['cannotsavemd5file'] = 'Није могуће сачувати md5 да
 $string['cannotsavezipfile'] = 'Није могуће сачувти ZIP архиву.';
 $string['cannotunzipfile'] = 'Није могуће распаковати ZIP датотеку.';
 $string['componentisuptodate'] = 'Компонента је доступна у својој најновијој верзији';
-$string['dmlexceptiononinstall'] = '<p>Дошло је до грешке везане за базу података [{$a->errorcode}].<br />{$a->debuginfo}</p>';
 $string['downloadedfilecheckfailed'] = 'Није успела провера преузете датотеке';
 $string['invalidmd5'] = 'Неисправна md5 датотека';
 $string['missingrequiredfield'] = 'Недостаје неко обавезно поље';
