diff --git a/install/lang/es_mx/error.php b/install/lang/es_mx/error.php
index d65d112..31096ef 100644
--- a/install/lang/es_mx/error.php
+++ b/install/lang/es_mx/error.php
@@ -30,7 +30,6 @@
 
 defined('MOODLE_INTERNAL') || die();
 
-$string['cannotcreatedboninstall'] = '<p>No puede crearse la BasedeDatos.</p> <p>La BasedeDatos especificada no existe y el usuario dado no tiene permiso para crear la BasedeDatos.</p> <p>El administrador del sitio debería verificar la configuración de la BasedeDatos.</p>';
 $string['cannotcreatelangdir'] = 'No se puede crear el directorio de idioma.';
 $string['cannotcreatetempdir'] = 'No se puede crear el directorio temp.';
 $string['cannotdownloadcomponents'] = 'No se puede descargar componentes';
@@ -40,7 +39,6 @@ $string['cannotsavemd5file'] = 'No se puede guardar el archivo md5';
 $string['cannotsavezipfile'] = 'No se puede guardar el archivo ZIP';
 $string['cannotunzipfile'] = 'No se puede descomprimir el archivo';
 $string['componentisuptodate'] = 'El componente está actualizado';
-$string['dmlexceptiononinstall'] = '<p>Ocurrió un error en la BasedeDatos[{$a->errorcode}].<br />{$a->debuginfo}</p>';
 $string['downloadedfilecheckfailed'] = 'Ha fallado la comprobación del archivo descargado';
 $string['invalidmd5'] = 'La variable de verificación MD5 es incorrecta no es valida - trate nuevamente';
 $string['missingrequiredfield'] = 'Falta algún campo necesario';
