diff --git a/admin/webservice/service_user_settings.php b/admin/webservice/service_user_settings.php
index 8a87fdb..18d0ef5 100644
--- a/admin/webservice/service_user_settings.php
+++ b/admin/webservice/service_user_settings.php
@@ -68,7 +68,7 @@ if ($usersettingsform->is_cancelled()) {
     //TODO: assign capability
 
     //display successful notification
-    $notification = $OUTPUT->notification(get_string('usersettingssaved', 'webservice'), 'notifysuccess');
+    $notification = $OUTPUT->notification(get_string('usersettingssaved', 'webservice'), 'success');
 }
 
 echo $OUTPUT->header();
