diff --git a/mod/assign/feedback/comments/locallib.php b/mod/assign/feedback/comments/locallib.php
index 5c51b00..9ed302c 100644
--- a/mod/assign/feedback/comments/locallib.php
+++ b/mod/assign/feedback/comments/locallib.php
@@ -94,10 +94,7 @@ class assign_feedback_comments extends assign_feedback_plugin {
                 $commenttext = $feedbackcomments->commenttext;
             }
         }
-        // Note that this handles the difference between empty and not in the quickgrading
-        // form at all (hidden column).
-        $newvalue = optional_param('quickgrade_comments_' . $userid, false, PARAM_TEXT);
-        return ($newvalue !== false) && ($newvalue != $commenttext);
+        return optional_param('quickgrade_comments_' . $userid, '', PARAM_TEXT) != $commenttext;
     }
 
 
@@ -176,11 +173,6 @@ class assign_feedback_comments extends assign_feedback_plugin {
     public function save_quickgrading_changes($userid, $grade) {
         global $DB;
         $feedbackcomment = $this->get_feedback_comments($grade->id);
-        $feedbackpresent = optional_param('quickgrade_comments_' . $userid, false, PARAM_TEXT) !== false;
-        if (!$feedbackpresent) {
-            // Nothing to save (e.g. hidden column).
-            return true;
-        }
         if ($feedbackcomment) {
             $feedbackcomment->commenttext = optional_param('quickgrade_comments_' . $userid, '', PARAM_TEXT);
             return $DB->update_record('assignfeedback_comments', $feedbackcomment);
