diff --git a/install/lang/hu/error.php b/install/lang/hu/error.php
index 2ab2c48..52a8c02 100644
--- a/install/lang/hu/error.php
+++ b/install/lang/hu/error.php
@@ -30,7 +30,6 @@
 
 defined('MOODLE_INTERNAL') || die();
 
-$string['cannotcreatedboninstall'] = '<p>Az adatbázis nem hozható létre.</p> <p>A megadott adatbázis nem létezik, a felhasználó pedig nem jogosult létrehozni egyet.</p> <p>A portál rendszergazdájának ellenőrizni kell az adatbázis-beállításokat.</p>';
 $string['cannotcreatelangdir'] = 'Nem hozható létre a lang könyvtár.';
 $string['cannotcreatetempdir'] = 'Nem hozható létre a temp könyvtár.';
 $string['cannotdownloadcomponents'] = 'Az összetevőket nem lehet letölteni.';
@@ -40,7 +39,6 @@ $string['cannotsavemd5file'] = 'Az md5 állományt nem lehet elmenteni.';
 $string['cannotsavezipfile'] = 'A tömörített állományt nem lehet elmenteni.';
 $string['cannotunzipfile'] = 'A tömörített állományt nem lehet kicsomagolni.';
 $string['componentisuptodate'] = 'Az összetevő a legújabb.';
-$string['dmlexceptiononinstall'] = '<p>Adatbázishiba történt [{$a->errorcode}].<br />{$a->debuginfo}</p>';
 $string['downloadedfilecheckfailed'] = 'A letöltött állomány ellenőrzése nem sikerült.';
 $string['invalidmd5'] = 'Az ellenőrző változó hibás volt – próbálkozzék ismét';
 $string['missingrequiredfield'] = 'Egy szükséges mező hiányzik.';
