diff --git a/lib/db/services.php b/lib/db/services.php
index fa8550d..ec515a8 100644
--- a/lib/db/services.php
+++ b/lib/db/services.php
@@ -871,8 +871,7 @@ $services = array(
             'moodle_user_get_users_by_courseid',
             'moodle_message_send_instantmessages',
             'core_course_get_contents',
-            'core_get_component_strings',
-            'core_calendar_get_calendar_events'),
+            'core_get_component_strings'),
         'enabled' => 0,
         'restrictedusers' => 0,
         'shortname' => MOODLE_OFFICIAL_MOBILE_SERVICE,
