diff --git a/lib/db/upgrade.php b/lib/db/upgrade.php
index 591f669..53acb74 100644
--- a/lib/db/upgrade.php
+++ b/lib/db/upgrade.php
@@ -85,7 +85,7 @@ defined('MOODLE_INTERNAL') || die();
  * @return bool always true
  */
 function xmldb_main_upgrade($oldversion) {
-    global $CFG, $USER, $DB, $OUTPUT, $SITE, $COURSE;
+    global $CFG, $USER, $DB, $OUTPUT, $SITE;
 
     require_once($CFG->libdir.'/db/upgradelib.php'); // Core Upgrade-related functions
 
@@ -316,10 +316,6 @@ function xmldb_main_upgrade($oldversion) {
             $dbman->drop_field($table, $field);
         }
 
-        // Since structure of 'course' table has changed we need to re-read $SITE from DB.
-        $SITE = $DB->get_record('course', array('id' => $SITE->id));
-        $COURSE = clone($SITE);
-
         upgrade_main_savepoint(true, 2012031500.02);
     }
 
@@ -451,10 +447,6 @@ function xmldb_main_upgrade($oldversion) {
             $dbman->add_field($table, $field);
         }
 
-        // Since structure of 'course' table has changed we need to re-read $SITE from DB.
-        $SITE = $DB->get_record('course', array('id' => $SITE->id));
-        $COURSE = clone($SITE);
-
         // Main savepoint reached
         upgrade_main_savepoint(true, 2012050300.03);
     }
@@ -571,10 +563,6 @@ function xmldb_main_upgrade($oldversion) {
             $dbman->add_field($table, $field);
         }
 
-        // Since structure of 'course' table has changed we need to re-read $SITE from DB.
-        $SITE = $DB->get_record('course', array('id' => $SITE->id));
-        $COURSE = clone($SITE);
-
         // Add course_sections_availability to add completion & grade availability conditions
         $table = new xmldb_table('course_sections_availability');
 
@@ -1348,10 +1336,6 @@ function xmldb_main_upgrade($oldversion) {
         // Launch change of type for field format
         $dbman->change_field_type($table, $field);
 
-        // Since structure of 'course' table has changed we need to re-read $SITE from DB.
-        $SITE = $DB->get_record('course', array('id' => $SITE->id));
-        $COURSE = clone($SITE);
-
         // Main savepoint reached
         upgrade_main_savepoint(true, 2012110200.00);
     }
@@ -1404,10 +1388,6 @@ function xmldb_main_upgrade($oldversion) {
             }
         }
 
-        // Since structure of 'course' table has changed we need to re-read $SITE from DB.
-        $SITE = $DB->get_record('course', array('id' => $SITE->id));
-        $COURSE = clone($SITE);
-
         // Main savepoint reached
         upgrade_main_savepoint(true, 2012110201.00);
     }
@@ -1524,7 +1504,6 @@ function xmldb_main_upgrade($oldversion) {
         if ($SITE->format !== 'site') {
             $DB->set_field('course', 'format', 'site', array('id' => $SITE->id));
             $SITE->format = 'site';
-            $COURSE->format = 'site';
         }
 
         // Main savepoint reached
@@ -2014,10 +1993,6 @@ function xmldb_main_upgrade($oldversion) {
             $dbman->drop_field($table, $field);
         }
 
-        // Since structure of 'course' table has changed we need to re-read $SITE from DB.
-        $SITE = $DB->get_record('course', array('id' => $SITE->id));
-        $COURSE = clone($SITE);
-
         // Main savepoint reached.
         upgrade_main_savepoint(true, 2013040300.01);
     }
@@ -2241,7 +2216,30 @@ function xmldb_main_upgrade($oldversion) {
     if ($oldversion < 2013051402.10) {
         // Find all fileareas that have missing root folder entry and add the root folder entry.
         if (empty($CFG->filesrootrecordsfixed)) {
-            upgrade_fix_missing_root_folders();
+            $sql = "SELECT distinct f1.contextid, f1.component, f1.filearea, f1.itemid
+                FROM {files} f1 left JOIN {files} f2
+                    ON f1.contextid = f2.contextid
+                    AND f1.component = f2.component
+                    AND f1.filearea = f2.filearea
+                    AND f1.itemid = f2.itemid
+                    AND f2.filename = '.'
+                    AND f2.filepath = '/'
+                WHERE (f1.component <> 'user' or f1.filearea <> 'draft')
+                and f2.id is null";
+            $rs = $DB->get_recordset_sql($sql);
+            $defaults = array('filepath' => '/',
+                            'filename' => '.',
+                            'userid' => $USER->id,
+                            'filesize' => 0,
+                            'timecreated' => time(),
+                            'timemodified' => time(),
+                            'contenthash' => sha1(''));
+            foreach ($rs as $r) {
+                $pathhash = sha1("/$r->contextid/$r->component/$r->filearea/$r->itemid".'/.');
+                $DB->insert_record('files', (array)$r + $defaults +
+                        array('pathnamehash' => $pathhash));
+            }
+            $rs->close();
             // To skip running the same script on the upgrade to the next major release.
             set_config('filesrootrecordsfixed', 1);
         }
@@ -2323,27 +2321,6 @@ function xmldb_main_upgrade($oldversion) {
         upgrade_main_savepoint(true, 2013051403.09);
     }
 
-    if ($oldversion < 2013051404.02) {
-        // Fix gradebook sortorder duplicates.
-        upgrade_grade_item_fix_sortorder();
-
-        // Main savepoint reached.
-        upgrade_main_savepoint(true, 2013051404.02);
-    }
-
-    if ($oldversion < 2013051404.04) {
-        // Remove deleted users home pages.
-        $sql = "DELETE FROM {my_pages}
-                WHERE EXISTS (SELECT {user}.id
-                                  FROM {user}
-                                  WHERE {user}.id = {my_pages}.userid
-                                  AND {user}.deleted = 1)
-                AND {my_pages}.private = 1";
-        $DB->execute($sql);
-
-        // Reached main savepoint.
-        upgrade_main_savepoint(true, 2013051404.04);
-    }
 
     return true;
 }
