diff --git a/mod/quiz/tests/editlib_test.php b/mod/quiz/tests/editlib_test.php
index a1b011d..1bcc14b 100644
--- a/mod/quiz/tests/editlib_test.php
+++ b/mod/quiz/tests/editlib_test.php
@@ -82,16 +82,4 @@ class mod_quiz_editlib_testcase extends basic_testcase {
         $this->assertEquals(quiz_add_page_break_at('1,2,0', 2), '1,2,0,0');
         $this->assertEquals(quiz_add_page_break_at('1,2,0', 3), '1,2,0');
     }
-
-    public function test_quiz_question_tostring() {
-        $question = new stdClass();
-        $question->qtype = 'multichoice';
-        $question->name = 'The question name';
-        $question->questiontext = '<p>What sort of <b>inequality</b> is x &lt; y<img alt="?" src="..."></p>';
-        $question->questiontextformat = FORMAT_HTML;
-
-        $summary = quiz_question_tostring($question);
-        $this->assertEquals('<span class="questionname">The question name</span>' .
-                '<span class="questiontext">What sort of INEQUALITY is x &lt; y[?]</span>', $summary);
-    }
 }
