diff --git a/backup/moodle2/restore_final_task.class.php b/backup/moodle2/restore_final_task.class.php
index 71371cc..dc49016 100644
--- a/backup/moodle2/restore_final_task.class.php
+++ b/backup/moodle2/restore_final_task.class.php
@@ -162,12 +162,7 @@ class restore_final_task extends restore_task {
         // rules from other tasks (activities) not belonging to one module instance (cmid = 0), so are restored here
         $rules = array_merge($rules, restore_logs_processor::register_log_rules_for_course());
 
-        // Calendar rules.
-        $rules[] = new restore_log_rule('calendar', 'add', 'event.php?action=edit&id={event}', '[name]');
-        $rules[] = new restore_log_rule('calendar', 'edit', 'event.php?action=edit&id={event}', '[name]');
-        $rules[] = new restore_log_rule('calendar', 'edit all', 'event.php?action=edit&id={event}', '[name]');
-
-        // TODO: Other logs like 'upload'... will go here
+        // TODO: Other logs like 'calendar', 'upload'... will go here
 
         return $rules;
     }
