diff --git a/backup/moodle2/restore_stepslib.php b/backup/moodle2/restore_stepslib.php
index 14c7ca7..14e1c35 100644
--- a/backup/moodle2/restore_stepslib.php
+++ b/backup/moodle2/restore_stepslib.php
@@ -2210,7 +2210,6 @@ class restore_calendarevents_structure_step extends restore_structure_step {
         $result = $DB->record_exists_sql($sql, $arg);
         if (empty($result)) {
             $newitemid = $DB->insert_record('event', $params);
-            $this->set_mapping('event', $oldid, $newitemid);
             $this->set_mapping('event_description', $oldid, $newitemid, $restorefiles);
         }
 
@@ -2820,12 +2819,6 @@ class restore_activity_grades_structure_step extends restore_structure_step {
         }
         // no need to save any grade_letter mapping
     }
-
-    public function after_restore() {
-        // Fix grade item's sortorder after restore, as it might have duplicates.
-        $courseid = $this->get_task()->get_courseid();
-        grade_item::fix_duplicate_sortorder($courseid);
-    }
 }
 
 
@@ -4047,9 +4040,6 @@ abstract class restore_questions_activity_structure_step extends restore_activit
 
         $data->questionusageid = $this->get_new_parentid($nameprefix . 'question_usage');
         $data->questionid      = $question->newitemid;
-        if (!property_exists($data, 'variant')) {
-            $data->variant = 1;
-        }
         $data->timemodified    = $this->apply_date_offset($data->timemodified);
 
         $newitemid = $DB->insert_record('question_attempts', $data);
