diff --git a/lib/editor/htmlarea/popups/insert_image_std.php b/lib/editor/htmlarea/popups/insert_image_std.php
index 1bb913a..fdd5d45 100644
--- a/lib/editor/htmlarea/popups/insert_image_std.php
+++ b/lib/editor/htmlarea/popups/insert_image_std.php
@@ -1,14 +1,104 @@
 <?php // $Id$
     require("../../../../config.php");
+    require_once $CFG->dirroot . '/file/repository/repository.class.php';
+
+
+/// Remote Learner Edit -- BEGIN
 
     $id = optional_param('id', SITEID, PARAM_INT);
     $httpsrequired = optional_param('httpsrequired', 0, PARAM_BOOL);
+    $shared = optional_param('shared', '', PARAM_ALPHA);
+    $userid = optional_param('userid', 0, PARAM_INT);
+    $uuid   = optional_param('uuid', '', PARAM_TEXT);
+    $ouuid  = optional_param('ouuid', '', PARAM_TEXT);
+    $alf    = optional_param('alf', 0, PARAM_INT);
+    $dd     = optional_param('dd', 0, PARAM_INT);
+
+    // If ouuid is set, set uuid to ouuid
+    if (!empty($ouuid)) {
+        $uuid = $ouuid;
+    }
+
+    require_login($id);
+
+    /// Make sure the plug-in is enabled and setup correctly.
+    if (isset($CFG->repository_plugins_enabled) && strstr($CFG->repository_plugins_enabled, 'alfresco')) {
+        if (!$repo = repository_factory::factory('alfresco')) {
+            debugging('Could not create repository object.', DEBUG_DEVELOPER);
+        }
+
+        // If we didn't get here from the drop-down...
+        if (empty($uuid) && empty($dd)) {
+            if ($uuid = $repo->get_repository_location($id, $userid, $shared, false)) {
+                redirect($CFG->wwwroot . '/lib/editor/htmlarea/popups/insert_image_std.php?id=' . $id .
+                         '&amp;userid=' . $userid . '&amp;shared=' . $shared . '&amp;uuid=' . $uuid . '&amp;alf=1' .
+                         '&amp;choose=', '', 0);
+            }
+
+            if ($uuid = $repo->get_default_browsing_location($id, $userid, $shared, false)) {
+               redirect($CFG->wwwroot . '/lib/editor/htmlarea/popups/insert_image_std.php?id=' . $id .
+                         '&amp;userid=' . $userid . '&amp;shared=' . $shared . '&amp;uuid=' . $uuid . '&amp;alf=1' .
+                         '&amp;choose=', '', 0);
+            }
+        }
+
+        // Clear out the previous repository location settings.
+        $repo->set_repository_location('', 0, 0, '', true);
+    }
+
+        // Determine if the current user actually has the capabilties to access this functionality and editing as well.
+    if (empty($alf)) {
+        if ($id == SITEID) {
+            $canmanagefiles = has_capability('moodle/course:managefiles', get_context_instance(CONTEXT_SYSTEM));
+        } else {
+            $canmanagefiles = has_capability('moodle/course:managefiles', get_context_instance(CONTEXT_COURSE, $id));
+        }
+    } else {
+        $canaccessfiles = false;
+        $canmanagefiles = false;
+        $canaccessorganization = false;
+        $canmanageorganization = false;
+
+        if (!empty($USER->access['rdef'])) {
+            foreach ($USER->access['rdef'] as $rdef) {
+                if ($canaccessfiles && $canmanagefiles &&
+                    $canaccessorganization && $canmanageorganization) {
+                    continue;
+                }
+
+                if (isset($rdef['block/repository:viewowncontent']) &&
+                          $rdef['block/repository:viewowncontent'] == CAP_ALLOW) {
+                    $canaccessfiles = true;
+                } else if (isset($rdef['block/repository:createowncontent']) &&
+                                 $rdef['block/repository:createowncontent'] == CAP_ALLOW) {
+                    $canmanagefiles = true;
+                }
+                if (isset($rdef['block/repository:vieworganizationcontent']) &&
+                          $rdef['block/repository:vieworganizationcontent'] == CAP_ALLOW) {
+                    $canaccessorganization = true;
+                }
+                if (isset($rdef['block/repository:createorganizationcontent']) &&
+                          $rdef['block/repository:createorganizationcontent'] == CAP_ALLOW) {
+                    $canmanageorganization = true;
+                }
+            }
+        }
+
+        if (!$canaccessfiles) {
+            $capabilityname = get_capability_string('block/repository:viewowncontent');
+            print_error('nopermissions', '', '', $capabilityname);
+        }
+    }
+
+/// Remote Learner Edit -- END
 
     require_course_login($id);
 
     @header('Content-Type: text/html; charset=utf-8');
 
-    if ($httpsrequired or (!empty($_SERVER['HTTPS']) and $_SERVER['HTTPS'] != 'off')) {
+    $upload_max_filesize = get_max_upload_file_size($CFG->maxbytes);
+
+    if (!empty($httpsrequired) or (!empty($_SERVER['HTTPS']) and $_SERVER['HTTPS'] != 'off')) {
         $url = preg_replace('|https?://[^/]+|', '', $CFG->wwwroot).'/lib/editor/htmlarea/';
     } else {
         $url = $CFG->wwwroot.'/lib/editor/htmlarea/';
@@ -36,8 +126,9 @@ function Init() {
       document.getElementById("f_url").value = param["f_url"];
       document.getElementById("f_alt").value = param["f_alt"] ? param["f_alt"] : alt;
       document.getElementById("f_border").value = parseInt(param["f_border"] || 0);
-      document.getElementById("f_vert").value = param["f_vert"] != -1 ? param["f_vert"] : 0;
-      document.getElementById("f_horiz").value = param["f_horiz"] != -1 ? param["f_horiz"] : 0;
+      document.getElementById("f_width").value = param["f_width"];
+      document.getElementById("f_height").value = param["f_height"];
+      window.ipreview.location.replace('preview.php?id='+ <?php print($id);?> +'&imageurl='+ param["f_url"]);
   }
   document.getElementById("f_url").focus();
 };
@@ -45,7 +136,7 @@ function Init() {
 function onOK() {
   var required = {
     "f_url": "<?php print_string("mustenterurl", "editor");?>",
-    "f_url": "<?php print_string("pleaseenteralt", "editor");?>"
+    "f_alt": "<?php print_string("pleaseenteralt", "editor");?>"
   };
   for (var i in required) {
     var el = document.getElementById(i);
@@ -57,7 +148,7 @@ function onOK() {
   }
   // pass data back to the calling window
   var fields = ["f_url", "f_alt", "f_align", "f_border",
-                "f_horiz", "f_vert"];
+                "f_width", "f_height"];
   var param = new Object();
   for (var i in fields) {
     var id = fields[i];
@@ -123,70 +214,90 @@ function onPreview() {
   win.focus();
   return false;
 };
+
+function checkvalue(elm,formname) {
+    var el = document.getElementById(elm);
+    if(!el.value) {
+        alert("Nothing to do!");
+        el.focus();
+        return false;
+    }
+}
+
+function submit_form(dothis) {
+    if(dothis == "delete") {
+        window.ibrowser.document.dirform.action.value = "delete";
+    }
+    if(dothis == "move") {
+        window.ibrowser.document.dirform.action.value = "move";
+    }
+    if(dothis == "zip") {
+        window.ibrowser.document.dirform.action.value = "zip";
+    }
+
+    window.ibrowser.document.dirform.submit();
+    return false;
+}
+
 //]]>
 </script>
 
 <style type="text/css">
 html, body {
-  background: ButtonFace;
-  color: ButtonText;
-  font: 11px Tahoma,Verdana,sans-serif;
-  margin: 0px;
-  padding: 0px;
+margin: 2px;
+background-color: rgb(212,208,200);
+font-family: Tahoma, Verdana, sans-serif;
+font-size: 11px;
 }
-body { padding: 5px; }
-table {
-  font: 11px Tahoma,Verdana,sans-serif;
+.title {
+background-color: #ddddff;
+padding: 5px;
+border-bottom: 1px solid black;
+font-family: Tahoma, sans-serif;
+font-weight: bold;
+font-size: 14px;
+color: black;
 }
-form p {
-  margin-top: 5px;
-  margin-bottom: 5px;
+td, input, select, button {
+font-family: Tahoma, Verdana, sans-serif;
+font-size: 11px;
 }
-.fl { width: 9em; float: left; padding: 2px 5px; text-align: right; }
-.fr { width: 6em; float: left; padding: 2px 5px; text-align: right; }
-fieldset { padding: 0px 10px 5px 5px; }
-select, input, button { font: 11px Tahoma,Verdana,sans-serif; }
 button { width: 70px; }
 .space { padding: 2px; }
-
-.title { background: #ddf; color: #000; font-weight: bold; font-size: 120%; padding: 3px 10px; margin-bottom: 10px;
-border-bottom: 1px solid black; letter-spacing: 2px;
-}
-form { padding: 0px; margin: 0px; }
+form { margin-bottom: 0px; margin-top: 0px; }
 </style>
 </head>
 <body onload="Init()">
-
-<div class="title"><?php print_string("insertimage","editor");?></div>
-
-<form action="" method="get">
-<table border="0" width="100%" style="padding: 0px; margin: 0px">
-  <tbody>
-
+  <div class="title"><?php print_string("insertimage","editor");?></div>
+  <div class="space"></div>
+  <div class="space"></div>
+  <div class="space"></div>
+  <form action="" method="get" id="first">
+    <table width="100%" border="0" cellspacing="0" cellpadding="0">
   <tr>
-    <td style="width: 7em; text-align: right"><?php print_string("imageurl","editor");?>:</td>
-    <td><input type="text" name="url" id="f_url" style="width:75%"
-      title="Enter the image URL here" />
-      <button name="preview" onclick="return onPreview()"
-      title="Preview the image in a new window"><?php print_string("preview","editor");?></button>
-    </td>
+        <td width="15%" align="right"><?php print_string("imageurl","editor");?>:</td>
+        <td width="60%"><input name="f_url" type="text" id="f_url" style="width: 100%;" /></td>
+        <td width="23%" align="center">
+          <button name="btnOK" type="button" id="btnOK" onclick="return onOK();"><?php print_string("ok","editor") ?></button></td>
   </tr>
   <tr>
-    <td style="width: 7em; text-align: right"><?php print_string("alternatetext","editor");?>:</td>
-    <td><input type="text" name="alt" id="f_alt" style="width:100%"
-      title="For browsers that don't support images" /></td>
+        <td align="right"><?php print_string("alternatetext","editor");?>:</td>
+        <td><input name="f_alt" type="text" id="f_alt" style="width: 100%;" /></td>
+        <td align="center">
+          <button name="btnCancel" type="button" id="btnCancel" onclick="return onCancel();"><?php print_string("cancel","editor") ?></button></td>
   </tr>
-  </tbody>
-</table>
-
-<p />
-<fieldset style="float: left; margin-left: 5px;">
-<legend><?php print_string("layout","editor");?></legend>
-
-<div class="space"></div>
-
-<div class="fl"><?php print_string("alignment","editor");?>:</div>
-<select size="1" name="align" id="f_align"
+    </table>
+      <div class="space"></div>
+    <table width="100%" border="0" cellspacing="0" cellpadding="0">
+      <tr>
+        <td width="32%" valign="top">
+        <fieldset><legend><?php print_string("layout","editor");?></legend>
+        <div class="space"></div>
+          <table width="100%" border="0" cellspacing="0" cellpadding="0">
+            <tr>
+              <td width="40%" align="right"><?php print_string("alignment","editor");?>:</td>
+              <td width="60%">
+                <select size="1" name="f_align" id="f_align"
   title="Positioning of this image">
   <option value=""                       ><?php print_string("notset","editor") ?></option>
   <option value="left"                   ><?php print_string("left","editor") ?></option>
@@ -198,42 +309,185 @@ form { padding: 0px; margin: 0px; }
   <option value="bottom"                 ><?php print_string("bottom","editor") ?></option>
   <option value="middle"                 ><?php print_string("middle","editor") ?></option>
   <option value="top"                    ><?php print_string("top","editor") ?></option>
-</select>
+                </select></td>
+      </tr>
+      <tr>
+              <td width="40%" align="right"><?php print_string("borderthickness","editor") ?>:</td>
+              <td width="60%">
+                <input type="text" id="f_border" name="f_border" size="5" /></td>
+            </tr>
+        </table><div class="space"></div>
+        </fieldset>&nbsp;</td>
+        <td width="34%" valign="top">
+        <fieldset><legend><?php print_string("size","editor");?></legend>
+      <div class="space"></div>
+          <table width="100%" border="0" cellspacing="0" cellpadding="0">
+            <tr>
+              <td width="40%" align="right"><?php print_string("width","editor");?>:</td>
+              <td width="60%">
+                <input name="f_width" type="text" id="f_width" size="5" />
+              </td>
+            </tr>
+            <tr>
+              <td width="40%" align="right"><?php print_string("height","editor");?>:</td>
+              <td width="60%">
+                <input name="f_height" type="text" id="f_height" size="5" /></td>
+            </tr>
+            </table><div class="space"></div>
+            </fieldset></td>
+    </tr>
+  </table></form>
+  <table width="100%" border="0" cellspacing="0" cellpadding="0">
+<?php
+
+/// Remote Learner Edit -- BEGIN
+
+/// Build an array of options for a navigation drop-down menu.
+
+    // Make sure the plug-in is enabled and setup correctly.
+    if (isset($CFG->repository_plugins_enabled) && strstr($CFG->repository_plugins_enabled, 'alfresco') && !empty($repo)) {
+        $default = '';
+        $opts = $repo->file_browse_options($id, $userid, $ouuid, $shared, '',
+                                           'lib/editor/htmlarea/popups/insert_image_std.php?alf=' . $alf,
+                                           'lib/editor/htmlarea/popups/insert_image_std.php?alf=0',
+                                           'lib/editor/htmlarea/popups/insert_image_std.php?alf=1', $default);
+
+        if (!empty($opts)) {
+            echo '<tr><td width="450" align="right">'. get_string('browsefilesfrom', 'repository') . ': ';
+            popup_form($CFG->wwwroot . '/', $opts, 'filepluginselect', $default, '');
+            echo '</td></tr>';
+        }
+    }
+/// Remote Learner Edit -- END
+
+
+?>
+    <tr>
+      <td width="55%" valign="top"><?php
+        print_string("filebrowser","editor");
+        echo "<br />";
 
-<p />
+/// Remote Learner Edit -- BEGIN
 
-<div class="fl"><?php print_string("borderthickness","editor");?>:</div>
-<input type="text" name="border" id="f_border" size="5"
-title="Leave empty for no border" />
+        // Display the Alfresco file browser within the IFRAME if we are accessing Alfresco files.
+        if (empty($alf)) {
+            echo "<iframe id=\"ibrowser\" name=\"ibrowser\" src=\"{$CFG->wwwroot}/lib/editor/htmlarea/coursefiles.php" .
+                 "?usecheckboxes=1&id=$id\" style=\"width: 100%; height: 200px;\"></iframe>";
+        } else {
+            $ref = urlencode($CFG->wwwroot . '/lib/editor/htmlarea/popups/insert_image_std.php?alf=1&amp;');
 
-<div class="space"></div>
+            echo '<iframe id="ibrowser" name="ibrowser" src="' . $CFG->wwwroot . '/file/repository/alfresco/' .
+                 'coursefiles.php?usecheckboxes=1&amp;id=' . $id . '&amp;shared=' . $shared . '&amp;userid=' .
+                 $userid . '&amp;uuid=' . $uuid . '&amp;ref=' . $ref . '" style="width: 100%; height: 200px;">' .
+                 '</iframe>';
+        }
 
-</fieldset>
 
-<fieldset style="float:right; margin-right: 5px;">
-<legend><?php print_string("spacing","editor");?></legend>
+/// Remote Learner Edit -- END
 
-<div class="space"></div>
+      ?>
+      </td>
+      <td width="45%" valign="top"><?php print_string("preview","editor");?>:<br />
+      <iframe id="ipreview" name="ipreview" src="<?php echo $url ?>blank.html" style="width: 100%; height: 200px;"></iframe>
+      </td>
+    </tr>
+  </table>
+    <table width="100%" border="0" cellspacing="0" cellpadding="0">
+      <tr>
+        <td width="55%"><div class="space"></div>
+        <?php if ($canmanagefiles) { ?>
+        <table border="0" cellpadding="2" cellspacing="0">
+          <tr><td><?php print_string("selection","editor");?>: </td>
+          <td><form id="idelete">
+          <input name="btnDelete" type="submit" id="btnDelete" value="<?php print_string("delete","editor");?>" onclick="return submit_form('delete');" /></form></td>
+<?php if (empty($alf)) { ?>
+          <td><form id="imove">
+          <input name="btnMove" type="submit" id="btnMove" value="<?php print_string("move","editor");?>" onclick="return submit_form('move');" /></td>
+<?php } ?>
+          <td><form id="izip">
+          <input name="btnZip" type="submit" id="btnZip" value="<?php print_string("zip","editor");?>" onclick="return submit_form('zip');" /></form></td>
+<?php if (empty($alf)) { ?>
+          <td><form method="post" action="../coursefiles.php" target="ibrowser">
+          <input type="hidden" name="id" value="<?php print($id);?>" />
+          <input type="hidden" name="wdir" value="" />
+          <input type="hidden" id="irename" name="file" value="" />
+          <input type="hidden" name="action" value="rename" />
+          <input type="hidden" name="sesskey" value="<?php p($USER->sesskey) ?>" />
+          <input name="btnRename" type="submit" id="btnRename" value="<?php print_string("rename","editor");?>" /></form></td>
+<?php } ?>
+          <tr></table>
+          <br />
+          <?php
+          } else {
+              print "";
+          } ?>
+        </td>
+      <td width="45%" rowspan="2" valign="top"><fieldset>
+          <legend><?php print_string("properties","editor");?></legend>
+          <div class="space"></div>
+          <div class="space"></div>
+          &nbsp;&nbsp;<?php print_string("size","editor");?>:
+          <input type="text" id="isize" name="isize" size="10" style="background: transparent; border: none;" />
+      <?php print_string("type","editor");?>: <input type="text" id="itype" name="itype" size="10" style="background: transparent; border: none;" />
+      <div class="space"></div>
+      <div class="space"></div>
+      </fieldset></td>
+    </tr>
+    <tr>
+      <td height="22">
+        <form id="cfolder" action="<?php echo $CFG->wwwroot . '/file/repository/alfresco/'; ?>coursefiles.php" method="post" target="ibrowser">
+          <input type="hidden" name="userid" value="<?php print($userid); ?>" />
+          <input type="hidden" name="id" value="<?php print($id);?>" />
+          <input type="hidden" name="wdir" value="" />
+          <input type="hidden" name="action" value="mkdir" />
+          <input type="hidden" name="sesskey" value="<?php p($USER->sesskey) ?>" />
+          <input name="name" type="text" id="foldername" size="35" />
+          <input name="btnCfolder" type="submit" id="btnCfolder" value="<?php print_string("createfolder","editor");?>" onclick="return checkvalue('foldername','cfolder');" />
+        </form>
+        <div class="space"></div>
+<?php
+
+    require_once($CFG->libdir . '/uploadlib.php');
+
+    // Build up the URL used in the upload form.
+    $vars = array(
+        'id'      => $id,
+        'shared'  => $shared,
+        'userid'  => $userid,
+        'uuid'    => $uuid,
+        'action'  => 'upload',
+        'sesskey' => $USER->sesskey,
+        'save'    => urlencode(get_string('upload', 'editor'))
+    );
+
+    $action = $CFG->wwwroot . '/file/repository/alfresco/coursefiles.php?';
+
+    $count = count($vars);
+    $i     = 0;
+    foreach ($vars as $var => $val) {
+        $action .= $var . '=' . $val . ($i < $count - 1 ? '&amp;' : '');
+        $i++;
+    }
 
-<div class="fr"><?php print_string("horizontal","editor");?>:</div>
-<input type="text" name="horiz" id="f_horiz" size="5"
-title="Horizontal padding" />
+    echo '<form action="' . $action . '" method="post" enctype="multipart/form-data" target="ibrowser" id="uploader">';
 
-<p />
+    // Include the form variables.
+    unset($vars['save']);
 
-<div class="fr"><?php print_string("vertical","editor");?>:</div>
-<input type="text" name="vert" id="f_vert" size="5"
-title="Vertical padding" />
+    foreach ($vars as $var => $val) {
+        echo '    <input type="hidden" name="' . $var . '" value="' . $val . '" />' . "\n";
+    }
 
-<div class="space"></div>
+    upload_print_form_fragment(1, array('userfile'), null, false, null, $upload_max_filesize, 0, false);
 
-</fieldset>
+    echo '<input name="save" type="submit" id="save" onclick="return checkvalue(\'userfile\',\'uploader\');" ' .
+         'value="' . get_string('upload', 'editor') . '" />';
 
-<div style="margin-top: 85px; text-align: right;">
-<hr />
-<button type="button" name="ok" onclick="return onOK();"><?php print_string("ok","editor");?></button>
-<button type="button" name="cancel" onclick="return onCancel();"><?php print_string("cancel","editor");?></button>
-</div>
-</form>
+?>
+        </form>
+      </td>
+    </tr>
+  </table>
+  <p>&nbsp;</p>
 </body>
 </html>
