diff --git a/lang/en/block.php b/lang/en/block.php
index 93d86b8..00527cd 100644
--- a/lang/en/block.php
+++ b/lang/en/block.php
@@ -50,7 +50,6 @@ $string['hidepanel'] = 'Hide panel';
 $string['moveblock'] = 'Move {$a} block';
 $string['moveblockafter'] = 'Move block to after {$a} block';
 $string['moveblockbefore'] = 'Move block to before {$a} block';
-$string['moveblockhere'] = 'Move block here';
 $string['movingthisblockcancel'] = 'Moving this block ({$a})';
 $string['onthispage'] = 'On this page';
 $string['pagetypes'] = 'Page types';
