diff --git a/lang/en/completion.php b/lang/en/completion.php
index 5d53c4b..6bb2073 100644
--- a/lang/en/completion.php
+++ b/lang/en/completion.php
@@ -108,7 +108,6 @@ $string['criteriarequiredany'] = 'Any criteria below are required';
 $string['csvdownload'] = 'Download in spreadsheet format (UTF-8 .csv)';
 $string['datepassed'] = 'Date passed';
 $string['days'] = 'Days';
-$string['daysoftotal'] = '{$a->days} of {$a->total}';
 $string['deletecompletiondata'] = 'Delete completion data';
 $string['dependencies'] = 'Dependencies';
 $string['dependenciescompleted'] = 'Completion of other courses';
