diff --git a/lang/en/dbtransfer.php b/lang/en/dbtransfer.php
index 0bd37d5..ca9a20f 100644
--- a/lang/en/dbtransfer.php
+++ b/lang/en/dbtransfer.php
@@ -35,5 +35,4 @@ $string['checkingsourcetables'] = 'Checking source table structure';
 $string['importschemaexception'] = 'Current database structure does not match all install.xml files. <br /> {$a}';
 $string['importversionmismatchexception'] = 'Current version {$a->currentver} does not match exported version {$a->schemaver}.';
 $string['malformedxmlexception'] = 'Malformed XML found, can not continue.';
-$string['tablex'] = 'Table {$a}:';
 $string['unknowntableexception'] = 'Unknown table {$a} found in export file.';
