diff --git a/lang/en/enrol.php b/lang/en/enrol.php
index 4cf053c..2458bdf 100644
--- a/lang/en/enrol.php
+++ b/lang/en/enrol.php
@@ -75,7 +75,7 @@ $string['expirynotifyall'] = 'Enroller and enrolled user';
 $string['expirynotifyenroller'] = 'Enroller only';
 $string['expirynotifyhour'] = 'Hour to send enrolment expiry notifications';
 $string['expirythreshold'] = 'Notification threshold';
-$string['expirythreshold_help'] = 'How long before enrolment expiry should users be notified?';
+$string['expirythreshold_help'] = 'How long before expiration should be users notified?';
 $string['finishenrollingusers'] = 'Finish enrolling users';
 $string['invalidenrolinstance'] = 'Invalid enrolment instance';
 $string['invalidrole'] = 'Invalid role';
