diff --git a/course/tests/behat/activities_visibility_icons.feature b/course/tests/behat/activities_visibility_icons.feature
index 56ccdf7..6ef447f 100644
--- a/course/tests/behat/activities_visibility_icons.feature
+++ b/course/tests/behat/activities_visibility_icons.feature
@@ -36,53 +36,3 @@ Feature: Toggle activities visibility from the course page
     And I log in as "student1"
     And I follow "Course 1"
     And "Test forum name" activity should be hidden
-
-  @javascript
-  Scenario: Activities can be shown and hidden inside a hidden section
-    Given the following "users" exists:
-      | username | firstname | lastname | email |
-      | teacher1 | Teacher | 1 | teacher1@asd.com |
-    And the following "courses" exists:
-      | fullname | shortname | format | numsections |
-      | Course 1 | C1 | topics | 2 |
-    And the following "course enrolments" exists:
-      | user | course | role |
-      | teacher1 | C1 | editingteacher |
-    And I log in as "teacher1"
-    And I follow "Course 1"
-    And I turn editing mode on
-    And I add a "Forum" to section "2" and I fill the form with:
-      | Forum name | Test forum name |
-      | Description | Test forum description |
-      | Visible | Show |
-    When I hide section "2"
-    Then "Test forum name" activity should be hidden
-    And I click on "Show" "link" in the "Test forum name" activity
-    And "Test forum name" activity should be visible
-    And I click on "Hide" "link" in the "Test forum name" activity
-    And "Test forum name" activity should be hidden
-
-  @javascript
-  Scenario: Activities can be shown and hidden inside an orphaned section
-    Given the following "users" exists:
-      | username | firstname | lastname | email |
-      | teacher1 | Teacher | 1 | teacher1@asd.com |
-    And the following "courses" exists:
-      | fullname | shortname | format | numsections |
-      | Course 1 | C1 | topics | 2 |
-    And the following "course enrolments" exists:
-      | user | course | role |
-      | teacher1 | C1 | editingteacher |
-    And I log in as "teacher1"
-    And I follow "Course 1"
-    And I turn editing mode on
-    And I add a "Forum" to section "2" and I fill the form with:
-      | Forum name | Test forum name |
-      | Description | Test forum description |
-      | Visible | Show |
-    When I click on ".reduce-sections" "css_element"
-    Then "Test forum name" activity should be visible
-    And I click on "Hide" "link" in the "Test forum name" activity
-    And "Test forum name" activity should be hidden
-    And I click on "Show" "link" in the "Test forum name" activity
-    And "Test forum name" activity should be visible
