diff --git a/course/tests/behat/section_visibility.feature b/course/tests/behat/section_visibility.feature
index ef4308d..eac2051 100644
--- a/course/tests/behat/section_visibility.feature
+++ b/course/tests/behat/section_visibility.feature
@@ -59,15 +59,11 @@ Feature: Show/hide course sections
     And section "3" should be hidden
     And I reload the page
     And section "1" should be hidden
-    And all activities in section "1" should be hidden
     And section "2" should be visible
     And section "3" should be hidden
-    And all activities in section "1" should be hidden
     And I log out
     And I log in as "student1"
     And I follow "Course 1"
     And section "1" should be hidden
-    And all activities in section "1" should be hidden
     And section "2" should be visible
     And section "3" should be hidden
-    And all activities in section "1" should be hidden
