diff --git a/theme/bootstrapbase/config.php b/theme/bootstrapbase/config.php
index 97003c0..d6eb2e9 100644
--- a/theme/bootstrapbase/config.php
+++ b/theme/bootstrapbase/config.php
@@ -39,6 +39,9 @@ $THEME->supportscssoptimisation = false;
 $THEME->editor_sheets = array('editor');
 
 $THEME->plugins_exclude_sheets = array(
+    'block' => array(
+        'html'
+    ),
     'gradereport' => array(
         'grader',
     ),
