diff --git a/lib/editor/tinymce/lib.php b/lib/editor/tinymce/lib.php
index 0d8736b..b4659b0 100644
--- a/lib/editor/tinymce/lib.php
+++ b/lib/editor/tinymce/lib.php
@@ -66,6 +66,9 @@ class tinymce_texteditor extends texteditor {
     }
 
     public function head_setup() {
+        global $CFG;
+        require_once($CFG->dirroot .'/repository/lib.php');
+        repository::include_fmfp_css();
     }
 
     public function use_editor($elementid, array $options=null, $fpoptions=null) {
