diff --git a/cache/stores/mongodb/lib.php b/cache/stores/mongodb/lib.php
index cdf764f..a6fd6a2 100644
--- a/cache/stores/mongodb/lib.php
+++ b/cache/stores/mongodb/lib.php
@@ -175,7 +175,7 @@ class cachestore_mongodb extends cache_store implements cache_is_configurable {
      * @return int
      */
     public static function get_supported_modes(array $configuration = array()) {
-        return self::MODE_APPLICATION;
+        return self::MODE_APPLICATION + self::MODE_SESSION;
     }
 
     /**
