diff --git a/mod/assignment/lib.php b/mod/assignment/lib.php
index 5bfc7f9..87f3084 100644
--- a/mod/assignment/lib.php
+++ b/mod/assignment/lib.php
@@ -1247,7 +1247,7 @@ class assignment_base {
         $tabindex = 1; //tabindex for quick grading tabbing; Not working for dropdowns yet
         add_to_log($course->id, 'assignment', 'view submission', 'submissions.php?id='.$this->cm->id, $this->assignment->id, $this->cm->id);
 
-        $PAGE->set_title($this->assignment->name);
+        $PAGE->set_title(format_string($this->assignment->name,true));
         $PAGE->set_heading($this->course->fullname);
         echo $OUTPUT->header();
 
