diff --git a/admin/tool/behat/tests/behat/data_generators.feature b/admin/tool/behat/tests/behat/data_generators.feature
index 0e95aeb..7237f4c 100644
--- a/admin/tool/behat/tests/behat/data_generators.feature
+++ b/admin/tool/behat/tests/behat/data_generators.feature
@@ -1,4 +1,4 @@
-@tool @tool_behat
+@tool @tool_behat @_only_local
 Feature: Set up contextual data for tests
   In order to write tests quickly
   As a developer
@@ -90,10 +90,10 @@ Feature: Set up contextual data for tests
     And I expand "Users" node
     And I follow "Permissions"
     And I select "Student (1)" from "Advanced role override"
-    Then "mod/forum:editanypost" capability has "Allow" permission
+    Then the "mod/forum:editanypost" field should match "1" value
     And I press "Cancel"
     And I select "Teacher (1)" from "Advanced role override"
-    And "mod/forum:replynews" capability has "Prevent" permission
+    And the "mod/forum:replynews" field should match "-1" value
     And I press "Cancel"
 
   Scenario: Add course enrolments
