diff --git a/admin/tool/behat/tests/behat/edit_permissions.feature b/admin/tool/behat/tests/behat/edit_permissions.feature
index a7935b9..b16b0e0 100644
--- a/admin/tool/behat/tests/behat/edit_permissions.feature
+++ b/admin/tool/behat/tests/behat/edit_permissions.feature
@@ -25,10 +25,10 @@ Feature: Edit capabilities
       | moodle/grade:managesharedforms | Prevent |
       | moodle/course:request | Prohibit |
     When I follow "Edit Teacher role"
-    Then "block/mnet_hosts:myaddinstance" capability has "Allow" permission
-    And "moodle/community:add" capability has "Not set" permission
-    And "moodle/grade:managesharedforms" capability has "Prevent" permission
-    And "moodle/course:request" capability has "Prohibit" permission
+    Then the "block/mnet_hosts:myaddinstance" field should match "1" value
+    And the "moodle/community:add" field should match "0" value
+    And the "moodle/grade:managesharedforms" field should match "-1" value
+    And the "moodle/course:request" field should match "-1000" value
 
   @javascript
   Scenario: Course capabilities overrides
@@ -41,9 +41,9 @@ Feature: Edit capabilities
       | mod/forum:editanypost | Prevent |
       | mod/forum:addquestion | Allow |
     When I select "Student (3)" from "Advanced role override"
-    Then "mod/forum:deleteanypost" capability has "Prohibit" permission
-    And "mod/forum:editanypost" capability has "Prevent" permission
-    And "mod/forum:addquestion" capability has "Allow" permission
+    Then the "mod/forum:deleteanypost" field should match "-1000" value
+    And the "mod/forum:editanypost" field should match "-1" value
+    And the "mod/forum:addquestion" field should match "1" value
 
   @javascript
   Scenario: Module capabilities overrides
@@ -60,6 +60,6 @@ Feature: Edit capabilities
       | mod/forum:editanypost | Prevent |
       | mod/forum:addquestion | Allow |
     When I select "Student (3)" from "Advanced role override"
-    Then "mod/forum:deleteanypost" capability has "Prohibit" permission
-    And "mod/forum:editanypost" capability has "Prevent" permission
-    And "mod/forum:addquestion" capability has "Allow" permission
+    Then the "mod/forum:deleteanypost" field should match "-1000" value
+    And the "mod/forum:editanypost" field should match "-1" value
+    And the "mod/forum:addquestion" field should match "1" value
