diff --git a/install/lang/pt/error.php b/install/lang/pt/error.php
index fea77d9..f44c59c 100644
--- a/install/lang/pt/error.php
+++ b/install/lang/pt/error.php
@@ -30,7 +30,6 @@
 
 defined('MOODLE_INTERNAL') || die();
 
-$string['cannotcreatedboninstall'] = '<p> Não é possível criar a base de dados. </p><p> A base de dados especificada não existe e o utilizador em questão não tem permissão para criar a base de dados. </p><p> O administrador do site deve verificar a configuração da base de dados. </p>';
 $string['cannotcreatelangdir'] = 'Não é possível criar a pasta de pacotes linguísticos';
 $string['cannotcreatetempdir'] = 'Não é possível criar a pasta de ficheiros temporários';
 $string['cannotdownloadcomponents'] = 'Não é possível descarregar os componentes.';
@@ -40,7 +39,6 @@ $string['cannotsavemd5file'] = 'Não é possível gravar o ficheiro md5';
 $string['cannotsavezipfile'] = 'Não é possível gravar o ficheiro ZIP';
 $string['cannotunzipfile'] = 'Não é possível descompactar o ficheiro ZIP';
 $string['componentisuptodate'] = 'O componente está atualizado.';
-$string['dmlexceptiononinstall'] = '<p>Ocorreu um erro de base de dados [{$a->errorcode}].<br />{$a->debuginfo}</p>';
 $string['downloadedfilecheckfailed'] = 'A verificação do ficheiro descarregado falhou.';
 $string['invalidmd5'] = 'A variável de verificação está errada - tente novamente.';
 $string['missingrequiredfield'] = 'Um dos campos obrigatórios está em falta';
