diff --git a/course/dndupload.js b/course/dndupload.js
index 9ddc26e..b1dbc63 100644
--- a/course/dndupload.js
+++ b/course/dndupload.js
@@ -102,24 +102,23 @@ M.course_dndupload = {
      * is available (or to explain why it is not available)
      */
     add_status_div: function() {
-        var Y = this.Y,
-            coursecontents = Y.one('#' + this.pagecontentid),
-            div,
-            handlefile = (this.handlers.filehandlers.length > 0),
-            handletext = false,
-            handlelink = false,
-            i = 0,
-            styletop,
-            styletopunit;
-
+        var coursecontents = document.getElementById(this.pagecontentid);
         if (!coursecontents) {
             return;
         }
 
-        div = Y.Node.create('<div id="dndupload-status"></div>').setStyle('opacity', '0.0');
-        coursecontents.insert(div, 0);
+        var div = document.createElement('div');
+        div.id = 'dndupload-status';
+        div.style.opacity = 0.0;
+        coursecontents.insertBefore(div, coursecontents.firstChild);
 
-        for (i = 0; i < this.handlers.types.length; i++) {
+        var Y = this.Y;
+        div = Y.one(div);
+        var handlefile = (this.handlers.filehandlers.length > 0);
+        var handletext = false;
+        var handlelink = false;
+        var i;
+        for (i=0; i<this.handlers.types.length; i++) {
             switch (this.handlers.types[i].identifier) {
                 case 'text':
                 case 'text/html':
@@ -142,20 +141,16 @@ M.course_dndupload = {
         }
         div.setContent(M.util.get_string($msgident, 'moodle'));
 
-        styletop = div.getStyle('top') || '0px';
-        styletopunit = styletop.replace(/^\d+/, '');
-        styletop = parseInt(styletop.replace(/\D*$/, ''), 10);
-
         var fadeanim = new Y.Anim({
             node: '#dndupload-status',
             from: {
                 opacity: 0.0,
-                top: (styletop - 30).toString() + styletopunit
+                top: '-30px'
             },
 
             to: {
                 opacity: 1.0,
-                top: styletop.toString() + styletopunit
+                top: '0px'
             },
             duration: 0.5
         });
