diff --git a/course/edit.php b/course/edit.php
index ebe4ab9..c549cc1 100644
--- a/course/edit.php
+++ b/course/edit.php
@@ -45,11 +45,8 @@ if ($id) { // editing course
         print_error('cannoteditsiteform');
     }
 
-    // Login to the course and retrieve also all fields defined by course format.
-    $course = get_course($id);
+    $course = course_get_format($id)->get_course();
     require_login($course);
-    $course = course_get_format($course)->get_course();
-
     $category = $DB->get_record('course_categories', array('id'=>$course->category), '*', MUST_EXIST);
     $coursecontext = context_course::instance($course->id);
     require_capability('moodle/course:update', $coursecontext);
