diff --git a/course/lib.php b/course/lib.php
index 1fb1421..bf88367 100644
--- a/course/lib.php
+++ b/course/lib.php
@@ -70,6 +70,7 @@ function make_log_url($module, $url) {
         case 'login':
         case 'lib':
         case 'admin':
+        case 'calendar':
         case 'category':
         case 'mnet course':
             if (strpos($url, '../') === 0) {
@@ -78,9 +79,6 @@ function make_log_url($module, $url) {
                 $url = "/course/$url";
             }
             break;
-        case 'calendar':
-            $url = "/calendar/$url";
-            break;
         case 'user':
         case 'blog':
             $url = "/$module/$url";
