diff --git a/course/moodleform_mod.php b/course/moodleform_mod.php
index c563107..2279c34 100644
--- a/course/moodleform_mod.php
+++ b/course/moodleform_mod.php
@@ -486,13 +486,12 @@ abstract class moodleform_mod extends moodleform {
         if ($this->_features->groupings or $this->_features->groupmembersonly) {
             //groupings selector - used for normal grouping mode or also when restricting access with groupmembersonly
             $options = array();
+            $options[0] = get_string('none');
             if ($groupings = $DB->get_records('groupings', array('courseid'=>$COURSE->id))) {
                 foreach ($groupings as $grouping) {
                     $options[$grouping->id] = format_string($grouping->name);
                 }
             }
-            collatorlib::asort($options);
-            $options = array(0 => get_string('none')) + $options;
             $mform->addElement('select', 'groupingid', get_string('grouping', 'group'), $options);
             $mform->addHelpButton('groupingid', 'grouping', 'group');
         }
