diff --git a/mod/forum/lib.php b/mod/forum/lib.php
index 7999fca..b29f642 100644
--- a/mod/forum/lib.php
+++ b/mod/forum/lib.php
@@ -3519,7 +3519,7 @@ function forum_print_post($post, $discussion, $forum, &$cm, $course, $ownpost=fa
     $output .= html_writer::tag('div', implode(' | ', $commandhtml), array('class'=>'commands'));
 
     // Output link to post if required
-    if ($link && forum_user_can_post($forum, $discussion, $USER, $cm, $course, $modcontext)) {
+    if ($link) {
         if ($post->replies == 1) {
             $replystring = get_string('repliesone', 'forum', $post->replies);
         } else {
