diff --git a/mod/forum/renderer.php b/mod/forum/renderer.php
index 8c71070..3463504 100644
--- a/mod/forum/renderer.php
+++ b/mod/forum/renderer.php
@@ -83,14 +83,9 @@ class mod_forum_renderer extends plugin_renderer_base {
      */
     public function subscriber_overview($users, $forum , $course) {
         $output = '';
-        $modinfo = get_fast_modinfo($course);
         if (!$users || !is_array($users) || count($users)===0) {
             $output .= $this->output->heading(get_string("nosubscribers", "forum"));
-        } else if (!isset($modinfo->instances['forum'][$forum->id])) {
-            $output .= $this->output->heading(get_string("invalidmodule", "error"));
         } else {
-            $cm = $modinfo->instances['forum'][$forum->id];
-            $canviewemail = in_array('email', get_extra_user_fields(context_module::instance($cm->id)));
             $output .= $this->output->heading(get_string("subscribersto","forum", "'".format_string($forum->name)."'"));
             $table = new html_table();
             $table->cellpadding = 5;
@@ -98,11 +93,7 @@ class mod_forum_renderer extends plugin_renderer_base {
             $table->tablealign = 'center';
             $table->data = array();
             foreach ($users as $user) {
-                $info = array($this->output->user_picture($user, array('courseid'=>$course->id)), fullname($user));
-                if ($canviewemail) {
-                    array_push($info, $user->email);
-                }
-                $table->data[] = $info;
+                $table->data[] = array($this->output->user_picture($user, array('courseid'=>$course->id)), fullname($user), $user->email);
             }
             $output .= html_writer::table($table);
         }
