diff --git a/mod/forum/view.php b/mod/forum/view.php
index 4b651ad..2283ad4 100644
--- a/mod/forum/view.php
+++ b/mod/forum/view.php
@@ -106,9 +106,9 @@
 
 /// Print header.
 
-    $PAGE->set_title($forum->name);
+    $PAGE->set_title(format_string($forum->name));
     $PAGE->add_body_class('forumtype-'.$forum->type);
-    $PAGE->set_heading($course->fullname);
+    $PAGE->set_heading(format_string($course->fullname));
 
     echo $OUTPUT->header();
 
