diff --git a/mod/quiz/report/default.php b/mod/quiz/report/default.php
index 143070a..cd7d487 100644
--- a/mod/quiz/report/default.php
+++ b/mod/quiz/report/default.php
@@ -64,7 +64,7 @@ abstract class quiz_default_report {
         global $PAGE, $OUTPUT;
 
         // Print the page header.
-        $PAGE->set_title($quiz->name);
+        $PAGE->set_title(format_string($quiz->name));
         $PAGE->set_heading($course->fullname);
         echo $OUTPUT->header();
     }
