diff --git a/lib/adminlib.php b/lib/adminlib.php
index 6195d6b..09df511 100644
--- a/lib/adminlib.php
+++ b/lib/adminlib.php
@@ -8445,7 +8445,7 @@ class admin_setting_devicedetectregex extends admin_setting {
     public function output_html($data, $query='') {
         global $OUTPUT;
 
-        $out  = html_writer::start_tag('table', array('class' => 'generaltable'));
+        $out  = html_writer::start_tag('table', array('border' => 1, 'class' => 'generaltable'));
         $out .= html_writer::start_tag('thead');
         $out .= html_writer::start_tag('tr');
         $out .= html_writer::tag('th', get_string('devicedetectregexexpression', 'admin'));
