diff --git a/lib/customcheckslib.php b/lib/customcheckslib.php
index 5a15553..7c63c50 100644
--- a/lib/customcheckslib.php
+++ b/lib/customcheckslib.php
@@ -15,16 +15,15 @@
 // along with Moodle.  If not, see <http://www.gnu.org/licenses/>.
 
 /**
- * This is a place to put custom environment checks, if there is not a better place.
+ * This is a one-line short description of the file
  *
  * This library contains a collection of functions able to perform
  * some custom checks executed by environmental tests (automatically
  * executed on install & upgrade and under petition in the admin block).
  *
- * Any function in this library gets a environment_results object passed in.
- * It must return:
+ * Any function in this library must return:
  * - null: if the test isn't relevant and must not be showed (ignored)
- * - the environment_results object that was passed in, with the status set to:
+ * - environment_results object with the status set to:
  *     - true: if passed
  *     - false: if failed
  *
