diff --git a/lib/environmentlib.php b/lib/environmentlib.php
index 73337b2..3d8a0df 100644
--- a/lib/environmentlib.php
+++ b/lib/environmentlib.php
@@ -1123,7 +1123,7 @@ class environment_results {
      */
     var $part;
     /**
-     * @var bool true means the test passed and all is OK. false means it failed.
+     * @var bool
      */
     var $status;
     /**
@@ -1180,11 +1180,11 @@ class environment_results {
     /**
      * Set the status
      *
-     * @param bool $testpassed true means the test passed and all is OK. false means it failed.
+     * @param boolean $status the status (true/false)
      */
-    function setStatus($testpassed) {
-        $this->status = $testpassed;
-        if ($testpassed) {
+    function setStatus($status) {
+        $this->status=$status;
+        if ($status) {
             $this->setErrorCode(NO_ERROR);
         }
     }
@@ -1274,7 +1274,7 @@ class environment_results {
     /**
      * Get the status
      *
-     * @return bool true means the test passed and all is OK. false means it failed.
+     * @return boolean result
      */
     function getStatus() {
         return $this->status;
