diff --git a/lib/moodlelib.php b/lib/moodlelib.php
index cefccbf..8dc423e 100644
--- a/lib/moodlelib.php
+++ b/lib/moodlelib.php
@@ -3183,8 +3183,8 @@ function require_logout() {
  */
 function require_course_login($courseorid, $autologinguest = true, $cm = NULL, $setwantsurltome = true, $preventredirect = false) {
     global $CFG, $PAGE, $SITE;
-    $issite = ((is_object($courseorid) and $courseorid->id == SITEID)
-          or (!is_object($courseorid) and $courseorid == SITEID));
+    $issite = (is_object($courseorid) and $courseorid->id == SITEID)
+          or (!is_object($courseorid) and $courseorid == SITEID);
     if ($issite && !empty($cm) && !($cm instanceof cm_info)) {
         // note: nearly all pages call get_fast_modinfo anyway and it does not make any
         // db queries so this is not really a performance concern, however it is obviously
@@ -4073,9 +4073,6 @@ function delete_user(stdClass $user) {
     // Remove users private keys.
     $DB->delete_records('user_private_key', array('userid' => $user->id));
 
-    // Remove users customised pages.
-    $DB->delete_records('my_pages', array('userid' => $user->id, 'private' => 1));
-
     // force logout - may fail if file based sessions used, sorry
     session_kill_user($user->id);
 
@@ -10743,10 +10740,6 @@ function get_performance_info() {
     $info['html'] .= '<span class="dbqueries">DB reads/writes: '.$info['dbqueries'].'</span> ';
     $info['txt'] .= 'db reads/writes: '.$info['dbqueries'].' ';
 
-    $info['dbtime'] = round($DB->perf_get_queries_time(), 5);
-    $info['html'] .= '<span class="dbtime">DB queries time: '.$info['dbtime'].' secs</span> ';
-    $info['txt'] .= 'db queries time: ' . $info['dbtime'] . 's ';
-
     if (function_exists('posix_times')) {
         $ptimes = posix_times();
         if (is_array($ptimes)) {
