diff --git a/lib/navigationlib.php b/lib/navigationlib.php
index 217144e..66be034 100644
--- a/lib/navigationlib.php
+++ b/lib/navigationlib.php
@@ -2056,7 +2056,7 @@ class global_navigation extends navigation_node {
      * @return bool
      */
     protected function load_for_user($user=null, $forceforcontext=false) {
-        global $DB, $CFG, $USER, $SITE;
+        global $DB, $CFG, $USER, $SITE, $COURSE; // RL EDIT: BJB130215 - ELIS-7127
 
         if ($user === null) {
             // We can't require login here but if the user isn't logged in we don't
@@ -2184,7 +2184,7 @@ class global_navigation extends navigation_node {
         }
 
         if ($iscurrentuser && has_capability('moodle/user:manageownfiles', context_user::instance($USER->id))) {
-            $url = new moodle_url('/user/files.php');
+            $url = new moodle_url('/user/files.php', array('courseid' => $COURSE->id)); // RL EDIT: BJB130215 - ELIS-7127: added courseid
             $usernode->add(get_string('myfiles'), $url, self::TYPE_SETTING);
         }
 
@@ -3680,6 +3680,34 @@ class settings_navigation extends navigation_node {
 
         }
 
+        // RL EDIT: BJB130215
+        $systemcontext = get_system_context();
+        $systemcapabilities = array(
+            'repository/elis_files:viewsitecontent',
+            'repository/elis_files:createsitecontent',
+        );
+
+        $coursecapabilities = array(
+            'repository/elis_files:viewcoursecontent',
+            'repository/elis_files:createcoursecontent',
+        );
+
+        // ELIS files
+        if (has_capability('repository/elis_files:view', $coursecontext) &&
+            (has_any_capability($systemcapabilities, $systemcontext) || has_any_capability($coursecapabilities, $coursecontext))) {
+            require_once($CFG->dirroot.'/repository/lib.php');
+            require_once($CFG->dirroot.'/repository/elis_files/lib.php');
+            if (repository_elis_files::is_repo_visible('elis_files')) {
+                $url = new moodle_url('/repository/filemanager.php', array(
+                           'course' => $course->id,
+                           'ctx_id' => $coursecontext->id,
+                           'sesskey' => sesskey()));
+                $coursenode->add(get_string('pluginname', 'repository_elis_files'), $url, self::TYPE_SETTING, null, 'elisfiles',
+                                 new pix_icon('i/files', ''));
+            }
+        }
+        // End RL EDIT
+
         // Switch roles
         $roles = array();
         $assumedrole = $this->in_alternative_role();
