diff --git a/lib/outputlib.php b/lib/outputlib.php
index e443511..e5ae617 100644
--- a/lib/outputlib.php
+++ b/lib/outputlib.php
@@ -786,25 +786,14 @@ class theme_config {
                         continue;
                     }
 
-                    // Add main stylesheet.
+                    $plugincontent = '';
                     $sheetfile = "$fulldir/styles.css";
                     if (is_readable($sheetfile)) {
                         $cssfiles['plugins'][$type.'_'.$plugin] = $sheetfile;
                     }
-
-                    // Create a list of candidate sheets from parents (direct parent last) and current theme.
-                    $candidates = array();
-                    foreach (array_reverse($this->parent_configs) as $parent_config) {
-                        $candidates[] = $parent_config->name;
-                    }
-                    $candidates[] = $this->name;
-
-                    // Add the sheets found.
-                    foreach ($candidates as $candidate) {
-                        $sheetthemefile = "$fulldir/styles_{$candidate}.css";
+                    $sheetthemefile = "$fulldir/styles_{$this->name}.css";
                         if (is_readable($sheetthemefile)) {
-                            $cssfiles['plugins'][$type.'_'.$plugin.'_'.$candidate] = $sheetthemefile;
-                        }
+                        $cssfiles['plugins'][$type.'_'.$plugin.'_'.$this->name] = $sheetthemefile;
                     }
                 }
             }
