diff --git a/lib/outputrequirementslib.php b/lib/outputrequirementslib.php
index 79da0c0..ed1fe87 100644
--- a/lib/outputrequirementslib.php
+++ b/lib/outputrequirementslib.php
@@ -420,7 +420,28 @@ class page_requirements_manager {
                                                         array('nofilesavailable', 'repository'), array('overwrite', 'repository'),
                                                         array('renameto', 'repository'), array('fileexists', 'repository'),
                                                         array('fileexistsdialogheader', 'repository'), array('fileexistsdialog_editor', 'repository'),
-                                                        array('fileexistsdialog_filemanager', 'repository')
+                                                        array('fileexistsdialog_filemanager', 'repository'),
+                                                        // Added for ELIS files
+                                                        array('errorupload', 'repository_elis_files'),
+                                                        array('actions', 'repository_elis_files'),
+                                                        array('createfolder', 'repository_elis_files'),
+                                                        array('uploadfiles', 'repository_elis_files'),
+                                                        array('advanced', 'repository_elis_files'),
+                                                        array('detailview', 'repository_elis_files'),
+                                                        array('strdelete', 'repository_elis_files'),
+                                                        array('deletecheckwarning', 'repository_elis_files'),
+                                                        array('move', 'repository_elis_files'),
+                                                        array('movefiles', 'repository_elis_files'),
+                                                        array('onlyincategories', 'repository_elis_files'),
+                                                        array('nocategoriesfound', 'repository_elis_files'),
+                                                        array('pleaseenterfoldername', 'repository_elis_files'),
+                                                        array('pleaseselectfilesfirst', 'repository_elis_files'),
+                                                        array('pleaseselectfolder', 'repository_elis_files'),
+                                                        array('uploadpopuptitle', 'repository_elis_files'),
+                                                        array('filesuploaded', 'repository_elis_files'),
+                                                        array('withselectedfiles', 'repository_elis_files'),
+                                                        array('jump', 'repository_elis_files'),
+                                                        array('entername', 'repository'),
                                                     ));
                     break;
                 case 'core_comment':
