diff --git a/lib/pluginlib.php b/lib/pluginlib.php
index 1869137..5f439cf 100644
--- a/lib/pluginlib.php
+++ b/lib/pluginlib.php
@@ -397,10 +397,6 @@ class plugin_manager {
             return 'svn';
         }
 
-        if (is_dir($pluginroot.'/.hg')) {
-            return 'mercurial';
-        }
-
         return false;
     }
 
@@ -1991,10 +1987,6 @@ class available_update_deployer {
             return 'svn';
         }
 
-        if (is_dir($pluginroot.'/.hg')) {
-            return 'mercurial';
-        }
-
         return false;
     }
 
