diff --git a/install/lang/en/error.php b/install/lang/en/error.php
index 1aa589c..5b8a665 100644
--- a/install/lang/en/error.php
+++ b/install/lang/en/error.php
@@ -46,8 +46,7 @@ $string['dmlexceptiononinstall'] = '<p>A database error has occurred [{$a->error
 $string['downloadedfilecheckfailed'] = 'Downloaded file check failed';
 $string['invalidmd5'] = 'The check variable was wrong - try again';
 $string['missingrequiredfield'] = 'Some required field is missing';
-$string['remotedownloaderror'] = '<p>The download of the component to your server failed. Please verify proxy settings; the PHP cURL extension is highly recommended.</p>
-<p>You must download the <a href="{$a->url}">{$a->url}</a> file manually, copy it to "{$a->dest}" in your server and unzip it there.</p>';
+$string['remotedownloaderror'] = 'Download of component to your server failed, please verify proxy settings, PHP cURL extension is highly recommended.<br /><br />You must download the <a href="{$a->url}">{$a->url}</a> file manually, copy it to "{$a->dest}" in your server and unzip it there.';
 $string['wrongdestpath'] = 'Wrong destination path';
 $string['wrongsourcebase'] = 'Wrong source URL base';
 $string['wrongzipfilename'] = 'Wrong ZIP file name';
