diff --git a/install/lang/ru/error.php b/install/lang/ru/error.php
index 3b76ec9..f92c150 100644
--- a/install/lang/ru/error.php
+++ b/install/lang/ru/error.php
@@ -30,7 +30,6 @@
 
 defined('MOODLE_INTERNAL') || die();
 
-$string['cannotcreatedboninstall'] = '<p> Невозможно создать базу данных. </p><p> Указанная база данных не существует, и данный пользователь не имеет разрешения на создание базы данных. </p><p> Администратор сайта  должен проверить конфигурацию базы данных. </p>';
 $string['cannotcreatelangdir'] = 'Не удается создать каталог языка';
 $string['cannotcreatetempdir'] = 'Не удается создать временный каталог';
 $string['cannotdownloadcomponents'] = 'Невозможно загрузить компоненты.';
@@ -40,7 +39,6 @@ $string['cannotsavemd5file'] = 'Не удалось сохранить MD5-фа
 $string['cannotsavezipfile'] = 'Не удалось сохранить ZIP-файл';
 $string['cannotunzipfile'] = 'Не удалось распаковать файл';
 $string['componentisuptodate'] = 'Компонент не нуждается в обновлении';
-$string['dmlexceptiononinstall'] = '<p>Произошла ошибка базы данных [{$a->errorcode}].<br />{$a->debuginfo}</p>';
 $string['downloadedfilecheckfailed'] = 'Ошибка проверки загруженного файла';
 $string['invalidmd5'] = 'Некорректная md5';
 $string['missingrequiredfield'] = 'Отсутствуют некоторые обязательные поля';
