diff --git a/admin/cron.php b/admin/cron.php
index e1a94b4..5c5e70a 100644
--- a/admin/cron.php
+++ b/admin/cron.php
@@ -529,6 +529,33 @@
     require_once($CFG->dirroot.'/elis/core/cron.php');
     elis_cron();
 
+    // Run the repository cron, if any
+    mtrace("\nRunning repository crons if required...");
+
+    if (file_exists($CFG->dirroot . '/file/repository/repository.class.php') &&
+        !empty($CFG->repository_plugins_enabled)) {
+
+        if (!($plugins = explode(',', $CFG->repository_plugins_enabled))) {
+            $plugins = array($CFG->repository);
+        }
+
+        require_once($CFG->dirroot . '/file/repository/repository.class.php');
+
+        foreach ($plugins as $p) {
+            $repository = repository_factory::factory($p);
+
+            if (method_exists($repository, 'cron')) {
+                $repository->cron();
+            }
+
+            if (!empty($repository->log)) {
+                mtrace($repository->log);
+            }
+
+            unset($repository);
+        }
+    }
+
     // run any customized cronjobs, if any
     // looking for functions in lib/local/cron.php
     if (file_exists($CFG->dirroot.'/local/cron.php')) {
