diff --git a/admin/index.php b/admin/index.php
index 66a29b2..2b0b302 100644
--- a/admin/index.php
+++ b/admin/index.php
@@ -487,6 +487,8 @@
 /// Check all ELIS report plugins and upgrade if necessary
     upgrade_plugins('elis', 'elis', "$CFG->wwwroot/$CFG->admin/index.php");
 
+/// Check all repository plugins and upgrade if necessary
+    upgrade_plugins('repository', 'file/repository', "$CFG->wwwroot/$CFG->admin/index.php");
 
 /// just make sure upgrade logging is properly terminated
     upgrade_log_finish();
