diff --git a/repository/upload/tests/behat/upload_file.feature b/repository/upload/tests/behat/upload_file.feature
index 217505c..1e028f7 100644
--- a/repository/upload/tests/behat/upload_file.feature
+++ b/repository/upload/tests/behat/upload_file.feature
@@ -1,4 +1,4 @@
-@repository @repository_upload @_only_local @_file_upload
+@repository @repository_upload @_only_local
 Feature: Upload files
   In order to add contents
   As a user
