diff --git a/repository/alfresco/lang/en/repository_alfresco.php b/repository/alfresco/lang/en/repository_alfresco.php
index fe75bd5..1a63fb1 100644
--- a/repository/alfresco/lang/en/repository_alfresco.php
+++ b/repository/alfresco/lang/en/repository_alfresco.php
@@ -31,9 +31,6 @@ $string['notitle'] = 'notitle';
 $string['password'] = 'Password';
 $string['pluginname_help'] = 'A plug-in for Alfresco CMS';
 $string['pluginname'] = 'Alfresco repository';
-$string['security_key_notice_message_small'] = 'Due to a recent security issue found in the Alfresco repository, we advice you to restart your Alfresco server.';
-$string['security_key_notice_message_subject'] = 'Alfresco repository security notice';
-$string['security_key_notice_message_content'] = 'A recent security issue was discovered when using external links to the Alfresco Moodle repository. Users were able to gain access to the accounts of other users on the Alfresco server through the use of information contained in these links (tokens). This feature has now been disabled, but it is possible that the tokens contained within these links still allow access to another user\'s account. For your own protection, it is important that you restart your Alfresco server in order to expire the tokens.';
 $string['soapmustbeenabled'] = 'SOAP extension must be enabled for alfresco plugin';
 $string['space'] = 'Space';
 $string['username'] = 'User name';
