diff --git a/elis/core/lang/en/elis_core.php b/elis/core/lang/en/elis_core.php
index fa1be1f..f8c95fe 100644
--- a/elis/core/lang/en/elis_core.php
+++ b/elis/core/lang/en/elis_core.php
@@ -79,8 +79,8 @@ $string['fld_courserole_help'] = '<h1>Course role filter</h1>
 <p>This filter allows you to filter users based the role they have assigned in the course
 specified by its shortname from a specified course category (if the shortname textbox is empty,
 the category is "any category" and the role is "any role" then the filter is not active).</p>';
-$string['fld_curriculum'] = 'Curriculum';
 $string['fld_credits'] = 'Credits';
+$string['fld_curriculum'] = 'Curriculum';
 $string['fld_description'] = 'Description';
 $string['fld_email'] = 'Email address';
 $string['fld_email_help'] = '<h1>Email address filter</h1>
@@ -136,7 +136,7 @@ The filter has the following options:</p>
 <li>ends with - this option allows only ID numbers that end with the text entered (if no text is entered, then the filter is disabled)</li>
 <li>is empty - this option allows only ID numbers that are equal to the empty string (the text entered is ignored)</li>
 </ul>';
-$string['fld_inactive'] = 'Inactive filter';
+$string['fld_inactive'] = 'Inactive';
 $string['fld_inactive_help'] = '<h1>Inactive filter</h1>
 <p>This filter allows you to filter users\' status as active, inactive or both.</p>';
 $string['fld_iscustom'] = 'Custom';
