diff --git a/mod/wiki/comments.php b/mod/wiki/comments.php
index d40d881..f10fccb 100644
--- a/mod/wiki/comments.php
+++ b/mod/wiki/comments.php
@@ -59,10 +59,6 @@ $course = $DB->get_record('course', array('id' => $cm->course), '*', MUST_EXIST)
 
 require_login($course, true, $cm);
 
-if (!wiki_user_can_view($subwiki, $wiki)) {
-    print_error('cannotviewpage', 'wiki');
-}
-
 add_to_log($course->id, 'wiki', 'comments', "comments.php?pageid=".$pageid, $pageid, $cm->id);
 
 /// Print the page header
