diff --git a/mod/wiki/create_form.php b/mod/wiki/create_form.php
index 97f4d07..be0ff9a 100644
--- a/mod/wiki/create_form.php
+++ b/mod/wiki/create_form.php
@@ -77,7 +77,6 @@ class mod_wiki_create_form extends moodleform {
                 $groupname = $groupinfo[$groupid];
                 $mform->addElement('static', 'groupdesciption', get_string('group'), $groupname);
                 $mform->addElement('hidden', 'groupinfo', $groupid);
-                $mform->setType('groupinfo', PARAM_INT);
             }
         }
 
