diff --git a/mod/wiki/edit_form.php b/mod/wiki/edit_form.php
index 088b7fb..2d34a6c 100644
--- a/mod/wiki/edit_form.php
+++ b/mod/wiki/edit_form.php
@@ -101,8 +101,10 @@ class mod_wiki_edit_form extends moodleform {
         $mform->setType('contentformat', PARAM_ALPHANUMEXT);
 
         if (!empty($CFG->usetags)) {
+            $tags = !isset($this->_customdata['tags'])?"":$this->_customdata['tags'];
             $mform->addElement('header', 'tagshdr', get_string('tags', 'tag'));
             $mform->addElement('tags', 'tags', get_string('tags'));
+            $mform->setDefault('tags', $tags);
             $mform->setType('tags', PARAM_TEXT);
         }
 
