diff --git a/mod/wiki/edit.php b/mod/wiki/edit.php
index 85f4ae2..fa9da8e 100644
--- a/mod/wiki/edit.php
+++ b/mod/wiki/edit.php
@@ -75,10 +75,7 @@ if (!empty($section) && !$sectioncontent = wiki_get_section_page($page, $section
 require_login($course, true, $cm);
 
 $context = context_module::instance($cm->id);
-
-if (!wiki_user_can_edit($subwiki)) {
-    print_error('cannoteditpage', 'wiki');
-}
+require_capability('mod/wiki:editpage', $context);
 
 if ($option == get_string('save', 'wiki')) {
     if (!confirm_sesskey()) {
