diff --git a/mod/wiki/files.php b/mod/wiki/files.php
index 932d494..239e2e0 100644
--- a/mod/wiki/files.php
+++ b/mod/wiki/files.php
@@ -78,11 +78,7 @@ $context = context_module::instance($cm->id);
 
 $PAGE->set_url('/mod/wiki/files.php', array('pageid'=>$pageid));
 require_login($course, true, $cm);
-
-if (!wiki_user_can_view($subwiki, $wiki)) {
-    print_error('cannotviewfiles', 'wiki');
-}
-
+$PAGE->set_context($context);
 $PAGE->set_title(get_string('wikifiles', 'wiki'));
 $PAGE->set_heading(get_string('wikifiles', 'wiki'));
 $PAGE->navbar->add(format_string(get_string('wikifiles', 'wiki')));
@@ -97,8 +93,12 @@ echo $renderer->tabs($page, $tabitems, $options);
 
 
 echo $OUTPUT->box_start('generalbox');
-echo $renderer->wiki_print_subwiki_selector($PAGE->activityrecord, $subwiki, $page, 'files');
-echo $renderer->wiki_files_tree($context, $subwiki);
+if (has_capability('mod/wiki:viewpage', $context)) {
+    echo $renderer->wiki_print_subwiki_selector($PAGE->activityrecord, $subwiki, $page, 'files');
+    echo $renderer->wiki_files_tree($context, $subwiki);
+} else {
+    echo $OUTPUT->notification(get_string('cannotviewfiles', 'wiki'));
+}
 echo $OUTPUT->box_end();
 
 if (has_capability('mod/wiki:managefiles', $context)) {
