diff --git a/mod/wiki/history.php b/mod/wiki/history.php
index 2a3bb41..6601022 100644
--- a/mod/wiki/history.php
+++ b/mod/wiki/history.php
@@ -59,11 +59,8 @@ if (!$cm = get_coursemodule_from_instance('wiki', $wiki->id)) {
 $course = $DB->get_record('course', array('id' => $cm->course), '*', MUST_EXIST);
 
 require_login($course, true, $cm);
-
-if (!wiki_user_can_view($subwiki, $wiki)) {
-    print_error('cannotviewpage', 'wiki');
-}
-
+$context = context_module::instance($cm->id);
+require_capability('mod/wiki:viewpage', $context);
 add_to_log($course->id, 'wiki', 'history', "history.php?pageid=".$pageid, $pageid, $cm->id);
 
 /// Print the page header
