diff --git a/mod/wiki/lock.php b/mod/wiki/lock.php
index 50a3f9e..97cd63c 100644
--- a/mod/wiki/lock.php
+++ b/mod/wiki/lock.php
@@ -68,9 +68,8 @@ if (!empty($section) && !$sectioncontent = wiki_get_section_page($page, $section
 
 require_login($course, false, $cm);
 
-if (!wiki_user_can_edit($subwiki)) {
-    print_error('cannoteditpage', 'wiki');
-}
+$context = context_module::instance($cm->id);
+require_capability('mod/wiki:editpage', $context);
 
 $wikipage = new page_wiki_lock($wiki, $subwiki, $cm);
 $wikipage->set_page($page);
