diff --git a/mod/wiki/prettyview.php b/mod/wiki/prettyview.php
index 8a1c0f9..e208ecf 100644
--- a/mod/wiki/prettyview.php
+++ b/mod/wiki/prettyview.php
@@ -53,9 +53,8 @@ if (!$wiki = wiki_get_wiki($subwiki->wikiid)) {
 
 require_login($course, true, $cm);
 
-if (!wiki_user_can_view($subwiki, $wiki)) {
-    print_error('cannotviewpage', 'wiki');
-}
+$context = context_module::instance($cm->id);
+require_capability('mod/wiki:viewpage', $context);
 
 $wikipage = new page_wiki_prettyview($wiki, $subwiki, $cm);
 
