diff --git a/mod/wiki/search.php b/mod/wiki/search.php
index 8329514..518195f 100644
--- a/mod/wiki/search.php
+++ b/mod/wiki/search.php
@@ -45,16 +45,12 @@ if (!$gid = groups_get_activity_group($cm)) {
     $gid = 0;
 }
 if (!$subwiki = wiki_get_subwiki_by_group($cm->instance, $gid)) {
-    print_error('incorrectsubwikiid', 'wiki');
+    return false;
 }
 if (!$wiki = wiki_get_wiki($subwiki->wikiid)) {
     print_error('incorrectwikiid', 'wiki');
 }
 
-if (!wiki_user_can_view($subwiki, $wiki)) {
-    print_error('cannotviewfiles', 'wiki');
-}
-
 $wikipage = new page_wiki_search($wiki, $subwiki, $cm);
 
 $wikipage->set_search_string($search, $searchcontent);
