diff --git a/mod/wiki/viewversion.php b/mod/wiki/viewversion.php
index ecbc700..ecdd650 100644
--- a/mod/wiki/viewversion.php
+++ b/mod/wiki/viewversion.php
@@ -60,10 +60,6 @@ $course = $DB->get_record('course', array('id' => $cm->course), '*', MUST_EXIST)
 
 require_login($course, true, $cm);
 
-if (!wiki_user_can_view($subwiki, $wiki)) {
-    print_error('cannotviewpage', 'wiki');
-}
-
 $wikipage = new page_wiki_viewversion($wiki, $subwiki, $cm);
 
 $wikipage->set_page($page);
