diff --git a/mod/lesson/lib.php b/mod/lesson/lib.php
index fb5fbd1..8133255 100644
--- a/mod/lesson/lib.php
+++ b/mod/lesson/lib.php
@@ -885,11 +885,7 @@ function lesson_pluginfile($course, $cm, $context, $filearea, $args, $forcedownl
         $fullpath = "/$context->id/mod_lesson/$filearea/$pageid/".implode('/', $args);
 
     } else if ($filearea === 'mediafile') {
-        if (count($args) > 1) {
-            // Remove the itemid when it appears to be part of the arguments. If there is only one argument
-            // then it is surely the file name. The itemid is sometimes used to prevent browser caching.
-            array_shift($args);
-        }
+        array_shift($args); // ignore itemid - caching only
         $fullpath = "/$context->id/mod_lesson/$filearea/0/".implode('/', $args);
 
     } else {
