diff --git a/mod/scorm/datamodels/aicclib.php b/mod/scorm/datamodels/aicclib.php
index 49cf121..c33fd3f 100644
--- a/mod/scorm/datamodels/aicclib.php
+++ b/mod/scorm/datamodels/aicclib.php
@@ -14,14 +14,6 @@
 // You should have received a copy of the GNU General Public License
 // along with Moodle.  If not, see <http://www.gnu.org/licenses/>.
 
-/**
- * functions used by AICC packages.
- *
- * @package    mod_scorm
- * @copyright 1999 onwards Roberto Pinna
- * @license    http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
- */
-
 function scorm_add_time($a, $b) {
     $aes = explode(':', $a);
     $bes = explode(':', $b);
@@ -114,14 +106,8 @@ function scorm_forge_cols_regexp($columns, $remodule='(".*")?,') {
     return $regexp;
 }
 
-/**
- * Sets up AICC packages
- * Called whenever package changes
- * @param object $scorm instance - fields are updated and changes saved into database
 
- * @return bool
- */
-function scorm_parse_aicc(&$scorm) {
+function scorm_parse_aicc($scorm) {
     global $DB;
 
     if ($scorm->scormtype == SCORM_TYPE_AICCURL) {
@@ -371,19 +357,10 @@ function scorm_parse_aicc(&$scorm) {
         }
     }
 
-    // Find first launchable object.
-    $sqlselect = 'scorm = ? AND '.$DB->sql_isnotempty('scorm_scoes', 'launch', false, true);
-    // We use get_records here as we need to pass a limit in the query that works cross db.
-    $scoes = $DB->get_records_select('scorm_scoes', $sqlselect, array($scorm->id), 'id', 'id', 0, 1);
-    if (!empty($scoes)) {
-        $sco = reset($scoes); // We only care about the first record - the above query only returns one.
-        $scorm->launch = $sco->id;
-    } else {
-        $scorm->launch = $launch;
-    }
-
     $scorm->version = 'AICC';
 
+    $scorm->launch = $launch;
+
     return true;
 }
 
