diff --git a/enrol/manual/externallib.php b/enrol/manual/externallib.php
index 7cec282..34de18f 100644
--- a/enrol/manual/externallib.php
+++ b/enrol/manual/externallib.php
@@ -109,7 +109,6 @@ class enrol_manual_external extends external_api {
             }
 
             // Check manual enrolment plugin instance is enabled/exist.
-            $instance = null;
             $enrolinstances = enrol_get_instances($enrolment['courseid'], true);
             foreach ($enrolinstances as $courseenrolinstance) {
               if ($courseenrolinstance->enrol == "manual") {
