diff --git a/lib/behat/form_field/behat_form_date_selector.php b/lib/behat/form_field/behat_form_date_selector.php
index 71c477c..2d86a76 100644
--- a/lib/behat/form_field/behat_form_date_selector.php
+++ b/lib/behat/form_field/behat_form_date_selector.php
@@ -25,11 +25,15 @@
 
 // NOTE: no MOODLE_INTERNAL test here, this file may be required by behat before including /config.php.
 
-require_once(__DIR__  . '/behat_form_group.php');
+require_once(__DIR__  . '/behat_form_select.php');
 
 /**
  * Date form field.
  *
+ * Simple extension of behat_form_select to allow date-type
+ * select fields to be filled like select elements instead of
+ * text elements.
+ *
  * This class will be refactored in case we are interested in
  * creating more complex formats to fill date and date-time fields.
  *
@@ -38,5 +42,4 @@ require_once(__DIR__  . '/behat_form_group.php');
  * @copyright  2013 David Monllaó
  * @license    http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
  */
-class behat_form_date_selector extends behat_form_group {
-}
+class behat_form_date_selector extends behat_form_select {}
