diff --git a/lib/behat/form_field/behat_form_selectyesno.php b/lib/behat/form_field/behat_form_selectyesno.php
index 73f8bef..3759a17 100644
--- a/lib/behat/form_field/behat_form_selectyesno.php
+++ b/lib/behat/form_field/behat_form_selectyesno.php
@@ -38,5 +38,4 @@ require_once(__DIR__  . '/behat_form_select.php');
  * @copyright  2013 David Monllaó
  * @license    http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
  */
-class behat_form_selectyesno extends behat_form_select {
-}
+class behat_form_selectyesno extends behat_form_select {}
