diff --git a/lib/behat/form_field/behat_form_checkbox.php b/lib/behat/form_field/behat_form_checkbox.php
index 69a9cbc..fb220d1 100644
--- a/lib/behat/form_field/behat_form_checkbox.php
+++ b/lib/behat/form_field/behat_form_checkbox.php
@@ -47,37 +47,18 @@ class behat_form_checkbox extends behat_form_field {
      */
     public function set_value($value) {
 
-        if (!empty($value) && !$this->field->isChecked()) {
-
             if (!$this->running_javascript()) {
                 $this->field->check();
                 return;
             }
 
+        if (!empty($value) && !$this->field->isChecked()) {
             // Check it if it should be checked and it is not.
             $this->field->click();
 
-            // Trigger the onchange event as triggered when 'checking' the checkbox.
-            $this->session->getDriver()->triggerSynScript(
-                $this->field->getXPath(),
-                "Syn.trigger('change', {}, {{ELEMENT}})"
-            );
-
         } else if (empty($value) && $this->field->isChecked()) {
-
-            if (!$this->running_javascript()) {
-                $this->field->uncheck();
-                return;
-            }
-
             // Uncheck if it is checked and shouldn't.
             $this->field->click();
-
-            // Trigger the onchange event as triggered when 'checking' the checkbox.
-            $this->session->getDriver()->triggerSynScript(
-                $this->field->getXPath(),
-                "Syn.trigger('change', {}, {{ELEMENT}})"
-            );
         }
     }
 
@@ -89,25 +70,4 @@ class behat_form_checkbox extends behat_form_field {
     public function get_value() {
         return $this->field->isChecked();
     }
-
-    /**
-     * Is it enabled?
-     *
-     * @param string $expectedvalue Anything !empty() is considered checked.
-     * @return bool
-     */
-    public function matches($expectedvalue = false) {
-
-        $ischecked = $this->field->isChecked();
-
-        // Any non-empty value provided means that it should be checked.
-        if (!empty($expectedvalue) && $ischecked) {
-            return true;
-        } else if (empty($expectedvalue) && !$ischecked) {
-            return true;
-        }
-
-        return false;
-    }
-
 }
