diff --git a/lib/behat/form_field/behat_form_date_time_selector.php b/lib/behat/form_field/behat_form_date_time_selector.php
index 19181cf..22c4bef 100644
--- a/lib/behat/form_field/behat_form_date_time_selector.php
+++ b/lib/behat/form_field/behat_form_date_time_selector.php
@@ -25,18 +25,21 @@
 
 // NOTE: no MOODLE_INTERNAL test here, this file may be required by behat before including /config.php.
 
-require_once(__DIR__  . '/behat_form_date_selector.php');
+require_once(__DIR__  . '/behat_form_select.php');
 
 /**
  * Date time form field.
  *
+ * Simple extension of behat_form_select to allow datetime-type
+ * select fields to be filled like select elements instead of
+ * text elements.
+ *
  * This class will be refactored in case we are interested in
- * creating more complex formats to fill date-time fields.
+ * creating more complex formats to fill date and date-time fields.
  *
  * @package    core_form
  * @category   test
  * @copyright  2013 David Monllaó
  * @license    http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
  */
-class behat_form_date_time_selector extends behat_form_date_selector {
-}
+class behat_form_date_time_selector extends behat_form_select {}
