diff --git a/group/delete.php b/group/delete.php
index 88430a5..b58bb0c 100644
--- a/group/delete.php
+++ b/group/delete.php
@@ -32,7 +32,6 @@ $groupids = required_param('groups', PARAM_SEQUENCE);
 $confirm = optional_param('confirm', 0, PARAM_BOOL);
 
 $PAGE->set_url('/group/delete.php', array('courseid'=>$courseid,'groups'=>$groupids));
-$PAGE->set_pagelayout('standard');
 
 // Make sure course is OK and user has access to manage groups
 if (!$course = $DB->get_record('course', array('id' => $courseid))) {
