diff --git a/files/renderer.php b/files/renderer.php
index eda9047..bbc729f 100644
--- a/files/renderer.php
+++ b/files/renderer.php
@@ -204,6 +204,7 @@ class core_files_renderer extends plugin_renderer_base {
                 <div class="{!}fp-btn-add"><a href="#"><img src="'.$this->pix_url('a/add_file').'" /> '.$straddfile.'</a></div>
                 <div class="{!}fp-btn-mkdir"><a href="#"><img src="'.$this->pix_url('a/create_folder').'" /> '.$strmakedir.'</a></div>
                 <div class="{!}fp-btn-download"><a href="#"><img src="'.$this->pix_url('a/download_all').'" /> '.$strdownload.'</a></div>
+                <div class="{!}fp-menu-jumpto"></div>
             </div>
             <div class="{!}fp-viewbar">
                 <a class="{!}fp-vb-icons" href="#"></a>
@@ -519,6 +520,10 @@ class core_files_renderer extends plugin_renderer_base {
      * @return string
      */
     private function fp_js_template_generallayout() {
+        // RL edit: add "executesearch" to the listing
+        // note that title text is set up for "refresh" as well, but dynamically in
+        // JS so that it only displayed for the ELIS Files repo plugin
+
         $rv = '
 <div class="file-picker fp-generallayout">
     <div class="fp-repo-area">
@@ -528,16 +533,19 @@ class core_files_renderer extends plugin_renderer_base {
     </div>
     <div class="fp-repo-items">
         <div class="fp-navbar">
-            <div>
+            <div class="fp-height">
                 <div class="{!}fp-toolbar">
                     <div class="{!}fp-tb-back"><a href="#">'.get_string('back', 'repository').'</a></div>
                     <div class="{!}fp-tb-search"><form></form></div>
+                    <div class="{!}fp-tb-advancedsearch" title="'.get_string('advanced').'"><a href="#"><img src="'.$this->pix_url('a/advancedsearch').'" /></a></div>
+                    <div class="{!}fp-tb-executesearch" title="'.get_string('search').'"><a href="#"><img src="'.$this->pix_url('a/executesearch').'" /></a></div>
                     <div class="{!}fp-tb-refresh"><a href="#"><img src="'.$this->pix_url('a/refresh').'" /></a></div>
                     <div class="{!}fp-tb-logout"><img src="'.$this->pix_url('a/logout').'" /><a href="#"></a></div>
                     <div class="{!}fp-tb-manage"><a href="#"><img src="'.$this->pix_url('a/setting').'" /> '.get_string('manageurl', 'repository').'</a></div>
                     <div class="{!}fp-tb-help"><a href="#"><img src="'.$this->pix_url('a/help').'" /> '.get_string('help').'</a></div>
                     <div class="{!}fp-tb-message"></div>
                 </div>
+                <div class="{!}fp-vb-menu-jumpto"></div>
                 <div class="{!}fp-viewbar">
                     <a class="{!}fp-vb-icons" href="#"></a>
                     <a class="{!}fp-vb-details" href="#"></a>
