diff --git a/lib/dml/moodle_database.php b/lib/dml/moodle_database.php
index b31d363..9c70843 100644
--- a/lib/dml/moodle_database.php
+++ b/lib/dml/moodle_database.php
@@ -91,8 +91,6 @@ abstract class moodle_database {
     protected $reads = 0;
     /** @var int The database writes (performance counter).*/
     protected $writes = 0;
-    /** @var float Time queries took to finish, seconds with microseconds.*/
-    protected $queriestime = 0;
 
     /** @var int Debug level. */
     protected $debug  = 0;
@@ -443,10 +441,7 @@ abstract class moodle_database {
         $logerrors = !empty($this->dboptions['logerrors']);
         $iserror   = ($error !== false);
 
-        $time = $this->query_time();
-
-        // Will be shown or not depending on MDL_PERF values rather than in dboptions['log*].
-        $this->queriestime = $this->queriestime + $time;
+        $time = microtime(true) - $this->last_time;
 
         if ($logall or ($logslow and ($logslow < ($time+0.00001))) or ($iserror and $logerrors)) {
             $this->loggingquery = true;
@@ -477,14 +472,6 @@ abstract class moodle_database {
     }
 
     /**
-     * Returns the time elapsed since the query started.
-     * @return float Seconds with microseconds
-     */
-    protected function query_time() {
-        return microtime(true) - $this->last_time;
-    }
-
-    /**
      * Returns database server info array
      * @return array Array containing 'description' and 'version' at least.
      */
@@ -538,7 +525,7 @@ abstract class moodle_database {
         if (!$this->get_debug()) {
             return;
         }
-        $time = $this->query_time();
+        $time = microtime(true) - $this->last_time;
         $message = "Query took: {$time} seconds.\n";
         if (CLI_SCRIPT) {
             echo $message;
@@ -2362,12 +2349,4 @@ abstract class moodle_database {
     public function perf_get_queries() {
         return $this->writes + $this->reads;
     }
-
-    /**
-     * Time waiting for the database engine to finish running all queries.
-     * @return float Number of seconds with microseconds
-     */
-    public function perf_get_queries_time() {
-        return $this->queriestime;
-    }
 }
