diff --git a/lib/dml/sqlite3_pdo_moodle_database.php b/lib/dml/sqlite3_pdo_moodle_database.php
index bb2f188..7391954 100644
--- a/lib/dml/sqlite3_pdo_moodle_database.php
+++ b/lib/dml/sqlite3_pdo_moodle_database.php
@@ -148,7 +148,7 @@ class sqlite3_pdo_moodle_database extends pdo_moodle_database {
         foreach ($rstables as $table) {
             $table = $table['name'];
             $table = strtolower($table);
-            if ($this->prefix !== false && $this->prefix !== '') {
+            if ($this->prefix !== '') {
                 if (strpos($table, $this->prefix) !== 0) {
                     continue;
                 }
